# $B$ decay to $J/\psi\, \pi\, K$  final state
### Exploratoty notebook
 * Calculation of the various amplitudes
 * A little bit of MC studies
 
The invesitgation led to the publication [arXiv:1712.02815](https://arxiv.org/abs/1712.02815)

In [1]:
# Add required packages if not installed yet
# Pkg.add("Plots")
# Pkg.add("DataFrames")
# Pkg.add("StatsBase")
# Pkg.add("QuadGK")
# Pkg.add("GR")

In [2]:
using Plots
gr()

Plots.GRBackend()

In [3]:
bw(s,msq,Γ)=1/(msq-s-1.0im*sqrt(msq)*Γ)
plot([
        x->real(bw(x,0.892^2,0.15)),
        x->imag(bw(x,0.892^2,0.15))
    ],
    linspace(0.4,1.3,100),
    lab = ["real", "imag"]
)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 1.2 
 
 
 -2 
 
 
 0 
 
 
 2 
 
 
 4 
 
 
 6 
 
<polyline clip-path="url(#clip6202)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 29.2591,180.71 34.8254,179.343 40.3916,177.934 45.9578,176.482 51.524,174.984 57.0902,173.441 62.6565,171.852 68.2227,170.214 73.7889,168.529 79.3551,166.796 
 84.9213,165.015 90.4876,163.187 96.0538,161.313 101.62,159.396 107.186,157.438 112.752,155.444 118.319,153.419 123.885,151.372 129.451,149.312 135.017,147.252 
 140.584,145.207 146.15,143.199 151.716,141.252 157.282,139.396 162.848,137.671 168.415,136.122 173.981,134.805 179.547,133.789 185.113,133.153 190.68,132.992 
 196.246,133.415 201.812,134.547 207.378,136.528 212.944,139.508 218.511,143.644 224.077,149.091 229.643,155.988 235.209,164.445 240.776,174.523 246.342,186.215 
 251.908,199.425 257.474,213.963 263.04,229.541 268.607,245.786 274.173,262.267 279.739,278.533 285.305,294.15 290.872,308.743 296.438,322.02 302.004,333.784 
 307.57,343.937 313.136,352.468 318.703,359.435 324.269,364.946 329.835,369.139 335.401,372.168 340.967,374.192 346.534,375.36 352.1,375.813 357.666,375.677 
 363.232,375.06 368.799,374.06 374.365,372.755 379.931,371.216 385.497,369.497 391.063,367.647 396.63,365.703 402.196,363.697 407.762,361.654 413.328,359.594 
 418.895,357.534 424.461,355.485 430.027,353.46 435.593,351.464 441.159,349.504 446.726,347.585 452.292,345.709 457.858,343.879 463.424,342.096 468.991,340.36 
 474.557,338.673 480.123,337.034 485.689,335.442 491.255,333.897 496.822,332.397 502.388,330.943 507.954,329.532 513.52,328.163 519.087,326.836 524.653,325.549 
 530.219,324.3 535.785,323.089 541.351,321.914 546.918,320.773 552.484,319.667 558.05,318.592 563.616,317.549 569.183,316.536 574.749,315.553 580.315,314.597 
 
 "/>
<polyline clip-path="url(#clip6202)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 29.2591,229.484 34.8254,228.425 40.3916,227.3 45.9578,226.103 51.524,224.83 57.0902,223.473 62.6565,222.025 68.2227,220.478 73.7889,218.825 79.3551,217.055 
 84.9213,215.159 90.4876,213.125 96.0538,210.941 101.62,208.592 107.186,206.065 112.752,203.341 118.319,200.403 123.885,197.231 129.451,193.803 135.017,190.095 
 140.584,186.082 146.15,181.736 151.716,177.028 157.282,171.927 162.848,166.402 168.415,160.42 173.981,153.951 179.547,146.965 185.113,139.441 190.68,131.362 
 196.246,122.724 201.812,113.539 207.378,103.845 212.944,93.7051 218.511,83.2224 224.077,72.5429 229.643,61.862 235.209,51.4268 240.776,41.5326 246.342,32.5122 
 251.908,24.7169 257.474,18.4891 263.04,14.1292 268.607,11.8626 274.173,11.811 279.739,13.9774 285.305,18.245 290.872,24.3929 296.438,32.1234 302.004,41.0951 
 307.57,50.9566 313.136,61.3737 318.703,72.049 324.269,82.7333 329.835,93.2287 335.401,103.387 340.967,113.103 346.534,122.312 352.1,130.976 357.666,139.081 
 363.232,146.631 368.799,153.64 374.365,160.133 379.931,166.136 385.497,171.682 391.063,176.802 396.63,181.527 402.196,185.889 407.762,189.917 413.328,193.638 
 418.895,197.079 424.461,200.262 430.027,203.21 435.593,205.944 441.159,208.48 446.726,210.837 452.292,213.028 457.858,215.069 463.424,216.971 468.991,218.746 
 474.557,220.404 480.123,221.956 485.689,223.408 491.255,224.769 496.822,226.047 502.388,227.246 507.954,228.375 513.52,229.437 519.087,230.437 524.653,231.381 
 530.219,232.272 535.785,233.113 541.351,233.909 546.918,234.663 552.484,235.377 558.05,236.053 563.616,236.696 569.183,237.306 574.749,237.885 580.315,238.437 
 
 "/>
 
 
 
 
 real 
 
 
 
 imag

## Integral over Dalitz plot

In [4]:
using QuadGK

In [5]:
# first, lambda
λ(x,y,z)=x^2+y^2+z^2-2*x*y-2*y*z-2*z*x
# second, integrate phase space 
function projectDalitzX(ampl,s1,s,m1sq,m2sq,m3sq)
    s1 < (sqrt(m2sq)+sqrt(m3sq))^2 && return 0
    s1 > (sqrt(s)-sqrt(m1sq))^2 && return 0
    ranges = m1sq+m2sq+(s-s1-m1sq)*(s1+m2sq-m3sq)/(2*s1)-sqrt(λ(s,s1,m1sq)*λ(s1,m2sq,m3sq))/(2*s1)*[-1.,1.]
    return quadgk(
        s3->abs(ampl(s1,s3,s,m1sq,m2sq,m3sq))^2,
        ranges[2],
        ranges[1])[1]/(8*π)^2/s;
end

projectDalitzX (generic function with 1 method)

In [6]:
# masses
mB = 5.279;
mJpsi = 3.686; # 3.01;
mπ = 0.139;
mK = 0.485;
mKs = 0.893;
ΓKs = 0.05;
mμ = 0.105;
# projection of the dalitz plot for the B to J/psi pi K
bw_amp(s1,s3,s,m1sq,m2sq,m3sq) = bw(s1,mKs^2,ΓKs)
phsp_s1(s1,s,m1sq,m2sq,m3sq) = sqrt(λ(s1,m2sq,m3sq)*λ(s,s1,m1sq))/s1/(8π)^2/s;
plot([
        e1->log(projectDalitzX(bw_amp,e1^2,mB^2,mJpsi^2,mπ^2,mK^2)),
        e1->log(abs(bw(e1^2,mKs^2,ΓKs))^2*phsp_s1(e1^2,mB^2,mJpsi^2,mπ^2,mK^2))
    ],
    linspace((mπ+mK)+0.01,(mB-mJpsi)-0.01,200))

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.8 
 
 
 1.0 
 
 
 1.2 
 
 
 1.4 
 
 
 -10 
 
 
 -8 
 
 
 -6 
 
 
 -4 
 
 
 -2 
 
<polyline clip-path="url(#clip6402)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.8488,244.957 38.5848,235.986 41.3208,229.217 44.0568,223.664 46.7928,218.878 49.5288,214.614 52.2648,210.725 55.0008,207.114 57.7369,203.716 60.4729,200.483 
 63.2089,197.377 65.9449,194.371 68.6809,191.443 71.4169,188.575 74.1529,185.752 76.8889,182.96 79.6249,180.188 82.361,177.427 85.097,174.667 87.833,171.9 
 90.569,169.118 93.305,166.315 96.041,163.482 98.777,160.613 101.513,157.701 104.249,154.738 106.985,151.718 109.721,148.633 112.457,145.475 115.193,142.236 
 117.929,138.908 120.665,135.48 123.401,131.944 126.137,128.288 128.873,124.5 131.609,120.568 134.345,116.476 137.081,112.211 139.817,107.754 142.553,103.087 
 145.289,98.1883 148.025,93.0369 150.761,87.6087 153.497,81.8793 156.233,75.8249 158.969,69.4252 161.705,62.6685 164.441,55.5603 167.177,48.1393 169.913,40.5032 
 172.649,32.8489 175.385,25.5257 178.121,19.0831 180.857,14.2572 183.593,11.811 186.329,12.2235 189.065,15.426 191.801,20.8434 194.537,27.6946 197.273,35.2786 
 200.009,43.0943 202.745,50.8323 205.481,58.3224 208.217,65.4833 210.953,72.2853 213.689,78.7275 216.425,84.8236 219.162,90.5947 221.898,96.0641 224.634,101.256 
 227.37,106.192 230.106,110.895 232.842,115.383 235.578,119.674 238.314,123.784 241.05,127.729 243.786,131.519 246.522,135.168 249.258,138.686 251.994,142.082 
 254.73,145.364 257.466,148.542 260.202,151.621 262.938,154.608 265.674,157.509 268.41,160.33 271.146,163.074 273.882,165.748 276.618,168.354 279.354,170.897 
 282.09,173.38 284.826,175.807 287.562,178.18 290.298,180.502 293.034,182.776 295.77,185.004 298.506,187.189 301.242,189.332 303.978,191.436 306.714,193.503 
 309.45,195.533 312.186,197.53 314.922,199.493 317.658,201.426 320.394,203.328 323.13,205.202 325.866,207.049 328.602,208.869 331.338,210.664 334.074,212.436 
 336.81,214.184 339.546,215.91 342.282,217.615 345.018,219.299 347.754,220.964 350.49,222.61 353.226,224.238 355.962,225.849 358.698,227.443 361.434,229.021 
 364.17,230.584 366.906,232.132 369.642,233.666 372.378,235.187 375.114,236.694 377.85,238.19 380.586,239.673 383.322,241.145 386.058,242.606 388.794,244.057 
 391.53,245.498 394.266,246.93 397.002,248.352 399.738,249.766 402.474,251.172 405.21,252.57 407.946,253.961 410.682,255.346 413.418,256.724 416.154,258.096 
 418.89,259.462 421.626,260.823 424.362,262.18 427.098,263.533 429.834,264.881 432.57,266.226 435.306,267.568 438.042,268.908 440.778,270.246 443.514,271.581 
 446.25,272.916 448.986,274.25 451.722,275.584 454.458,276.918 457.194,278.253 459.93,279.589 462.666,280.926 465.402,282.267 468.139,283.61 470.875,284.957 
 473.611,286.308 476.347,287.665 479.083,289.027 481.819,290.395 484.555,291.771 487.291,293.155 490.027,294.549 492.763,295.952 495.499,297.367 498.235,298.794 
 500.971,300.235 503.707,301.691 506.443,303.163 509.179,304.654 511.915,306.165 514.651,307.698 517.387,309.255 520.123,310.839 522.859,312.453 525.595,314.099 
 528.331,315.782 531.067,317.506 533.803,319.275 536.539,321.094 539.275,322.971 542.011,324.913 544.747,326.928 547.483,329.027 550.219,331.224 552.955,333.535 
 555.691,335.981 558.427,338.587 561.163,341.389 563.899,344.432 566.635,347.783 569.371,351.535 572.107,355.836 574.843,360.927 577.579,367.257 580.315,375.813 
 
 "/>
<polyline clip-path="url(#clip6402)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.8488,244.957 38.5848,235.986 41.3208,229.217 44.0568,223.664 46.7928,218.878 49.5288,214.614 52.2648,210.725 55.0008,207.114 57.7369,203.716 60.4729,200.483 
 63.2089,197.377 65.9449,194.371 68.6809,191.443 71.4169,188.575 74.1529,185.752 76.8889,182.96 79.6249,180.188 82.361,177.427 85.097,174.667 87.833,171.9 
 90.569,169.118 93.305,166.315 96.041,163.482 98.77

## Phase space MonteCarlo

In [7]:
# prepare array of weights
nbinsX = 300;
ms3_bins = linspace((mJpsi+mπ)^2,(mB-mK)^2,nbinsX+1)
density(s3) = sqrt(λ(mB^2,s3,mK^2)*λ(s3,mJpsi^2,mπ^2))/s3
weights = [density((ms3_bins[i]+ms3_bins[i+1])/2) for i in 1:nbinsX];

In [8]:
using StatsBase

In [9]:
nEvents = 1000000;
inds = StatsBase.sample(1:nbinsX, Weights(weights), nEvents)
s3_data = [ms3_bins[i]+rand()*(ms3_bins[i+1]-ms3_bins[i]) for i in inds];

In [10]:
histogram(s3_data, bins=200)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 16 
 
 
 18 
 
 
 20 
 
 
 22 
 
 
 0 
 
 
 2000 
 
 
 4000 
 
 
 6000 
 
 
 8000 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1

In [11]:
using DataFrames

In [12]:
### Kinematic function
λ(x,y,z) = x^2+y^2+z^2-2*x*y-2*y*z-2*z*x;
density(s1,s,m1sq,m2sq,m3sq) = sqrt(λ(s,s1,m1sq)*λ(s1,m2sq,m3sq))/s1
s3_ranges(s1,s,m1sq,m2sq,m3sq) = m1sq+m2sq+(s-s1-m1sq)*(s1+m2sq-m3sq)/(2*s1)-sqrt(λ(s,s1,m1sq)*λ(s1,m2sq,m3sq))/(2*s1)*[1.,-1.]

### Functions to boost and rotate
Rz(θ) = [cos(θ) -sin(θ) 0 ; sin(θ) cos(θ) 0 ; 0 0 1];
Ry(θ) = [cos(θ) 0 sin(θ); 0 1 0 ; -sin(θ) 0 cos(θ)];
function Λz(γ)
    β = sqrt(1-1/γ^2);
    [γ 0 0 β*γ; 0 1 0 0; 0 0 1 0; β*γ 0 0 γ]
end
Rot4(R3) = [1 [0 0 0]; [0; 0; 0] R3];
boost_rotate(γ,θ,ϕ) = Rot4(Rz(ϕ)*Ry(θ))*Λz(γ);

# scalar products
mprod(P1,P2) = P1[1]*P2[1]-P1[2:4]⋅P2[2:4]
mnorm(P1) = mprod(P1,P1);

In [13]:
nEvents = 100000;

# data storage
tree = DataFrame(
    [Array{Float64,1} for i in 1:6],
    [:P1, :P2, :P3, :P4, :Q1, :Q2],
    nEvents);

# s-data
# prepare array of weights
nbinsX = 300;
ms_bins = linspace((mJpsi+mπ)^2,(mB-mK)^2,nbinsX+1)
density(s) = sqrt(λ(mB^2,s,mK^2)*λ(s,mJpsi^2,mπ^2))/s
weights = [density((ms_bins[i]+ms_bins[i+1])/2) for i in 1:nbinsX];
# generate nEvents-s-values
inds = StatsBase.sample(1:nbinsX, Weights(weights), nEvents)
s_data = [ms3_bins[i]+rand()*(ms3_bins[i+1]-ms3_bins[i]) for i in inds];

# main look to fill trees with four-vectors
mπsq = mπ^2;
for i in 1:nEvents

    s = s_data[i];
    
    # P2-frame
    cosT4 = 2*rand()-1.; sinT4 = sqrt(1-cosT4^2); θ4 = acos(cosT4);
    phi4 = π*(2*rand()-1.);
    p4 = sqrt(λ(mB^2, s, mK^2))/(2*mB);
    e4 = sqrt(p4^2+mK^2);  e13 = sqrt(p4^2+s);
    tree[i,:P2] = [mB,0,0,0];
    tree[i,:P4] = [e4,-p4*sinT4*cos(phi4),-p4*sinT4*sin(phi4),-p4*cosT4];

    # from P1-P3 rest frame
    γ13 = e13/sqrt(s);
    cosT1 = 2*rand()-1.; sinT1 = sqrt(1-cosT1^2); θ1 = acos(cosT1);
    phi1 = π*(2*rand()-1.);
    p1 = sqrt(λ(s, mJpsi^2, mπ^2)/(4*s));
    e1 = sqrt(p1^2+mJpsi^2); e3 = sqrt(p1^2+mπ^2);
    tree[i,:P1] = boost_rotate(γ13,θ4,phi4)*[e1,p1*sinT1*cos(phi1),p1*sinT1*sin(phi1),p1*cosT1];
    tree[i,:P3] = boost_rotate(γ13,θ4,phi4)*[e3,-p1*sinT1*cos(phi1),-p1*sinT1*sin(phi1),-p1*cosT1];
    
    # from Q1-Q2 rest frame
    γ1 = e1/mJpsi;
    cosTμ = 2*rand()-1.; sinTμ = sqrt(1-cosTμ^2);
    phiμ = π*(2*rand()-1.);
    pμ = sqrt(λ(mJpsi^2, mμ^2, mμ^2))/(2*mJpsi);
    eμ = sqrt(pμ^2+mμ^2);
    tree[i,:Q1] = boost_rotate(γ13,θ4,phi4)*boost_rotate(γ1,θ1,phi1)*[eμ, pμ*sinTμ*cos(phiμ), pμ*sinTμ*sin(phiμ), pμ*cosTμ];
    tree[i,:Q2] = boost_rotate(γ13,θ4,phi4)*boost_rotate(γ1,θ1,phi1)*[eμ,-pμ*sinTμ*cos(phiμ),-pμ*sinTμ*sin(phiμ),-pμ*cosTμ];    
end

In [14]:
tree[1,:]

1×6 DataFrames.DataFrame
│ Row │ P1                                     │ P2                     │
├─────┼────────────────────────────────────────┼────────────────────────┤
│ 1   │ [3.76595, 0.424746, -0.45863, 0.45281] │ [5.279, 0.0, 0.0, 0.0] │

│ Row │ P3                                        │
├─────┼───────────────────────────────────────────┤
│ 1   │ [0.653287, -0.343842, 0.474963, 0.252281] │

│ Row │ P4                                            │
├─────┼───────────────────────────────────────────────┤
│ 1   │ [0.859762, -0.0809043, -0.0163327, -0.705092] │

│ Row │ Q1                                       │
├─────┼──────────────────────────────────────────┤
│ 1   │ [2.01619, 0.132594, -1.97649, -0.360438] │

│ Row │ Q2                                     │
├─────┼────────────────────────────────────────┤
│ 1   │ [1.74976, 0.292153, 1.51786, 0.813248] │

In [15]:
# check if s-distribution is the same as input
histogram([
        [mnorm(tree[i,:P1]+tree[i,:P3]) for i=1:nEvents],
        s_data
        ],
     nbins=200,fillalpha=0.7)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 16 
 
 
 18 
 
 
 20 
 
 
 22 
 
 
 0 
 
 
 200 
 
 
 400 
 
 
 600 
 
 
 800 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 y1 
 
 
 
 
 y2

In [16]:
# check energy conservation
histogram([tree[i,:P2][1]-tree[i,:P3][1]-tree[i,:P4][1]-tree[i,:Q1][1]-tree[i,:Q2][1] for i=1:nEvents],
    nbins=200,
    xlims=(-0.1,0.1),lab="eB-sum_e")
# check B-invariant mass
histogram!([mB^2 - mnorm(tree[i,:P4]+tree[i,:P3]+tree[i,:Q1]+tree[i,:Q2]) for i=1:nEvents],
    nbins=200,
xlims=(-0.1,0.1),lab="mB^2-|sumP|^2")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 -0.10 
 
 
 -0.05 
 
 
 0.00 
 
 
 0.05 
 
 
 0.10 
 
 
 0 
 
 
 10000 
 
 
 20000 
 
 
 30000 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 eB-sum_e 
 
 
 
 
 mB^2-|sumP|^2

In [17]:
# check dalitz plot
histogram2d(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    [mnorm(tree[i,:Q1]+tree[i,:Q2]+tree[i,:P3]) for i in 1:nEvents],
    bins=(100,100),
    xlabel="t [pi Jpsi] (GeV)",
    ylabel="s [pi K] (GeV)",
    title = "Jpsi pi K phase space Dalitz plot"
    )

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 16 
 
 
 18 
 
 
 20 
 
 
 22 
 
 
 Jpsi pi K phase space Dalitz plot 
 
 
 t [pi Jpsi] (GeV) 
 
 
 s [pi K] (GeV) 
 
 
<image width="476" height="329" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAdwAAAFJCAYAAAAxCJwFAAAgAElEQVR4nO29e7yXdZn9vUwYOey9
SUHkpCQoGx4MRUCRUhBNdDh4GkXCSTooHnEm/WWWpTWm1aCT5hE7oEGKjkgoP8VBBC1EATGCgQ0K
IUeVrcneAo5Yzx+9nufFta41fO+o7jTX+79r87m/3/t8873XZ61rLxizRzT5w65V25Z9w7++uWNV
WuJT+5wZ6n/u/Ps0ZsXWFrv91uomeZmGnR8L9fDO69KYu5cfGD+nafz3R955IS1zSZu+6W/M7VsW
hrrjH7qG+jfvPpGW2fn+2/yXvSp+kTHmQ8/HKg8xxhhjzJ+LH7jGGGNMCfiBa4wxxpSAtaOPPFGL
BYDh1WNDXVuzd1pqVO3KUG9+++Ohfnpz24rLrNpyQBpT3fTdUC/c0iaNYcaNmBHqJYuOSGNWvdU6
1FVN3wt13dvVaRnWhq8f99M0Zvv6uH63Th+623UDgImPDwl1ddOdacwv1u8T6vXv/y7UNWiZlln3
sQ2hXt04w9e3MR8g/AvXGGOMKQE/cI0xxpgS8APXGGOMKYEmf+sVMH9Jdu+NBYDHj45a6zOvfiKN
6daqPtR3rWydxize1CnUh+4bl2mIEimArPMurG+VxrCHtrZVQ6jvfDlqmwBQO/e4UG/a1jyNYZ30
0DavhVpquLQNd94zOo3p1ipqq7Pf+J9Q15BeCwAbtv9D/APXAO4b82ioN73SOS6yo1lapnFH3O7G
d89J+vz0dVFb/3H95PQ5BzXvH+rVjTNphH3DxuwJ/oVrjDHGlIAfuMYYY0wJ+IFrjDHGlIC1mA8J
Vc26JT3uR92OCXWvTmtD/ZNf90qf071mW6h7t1+fxrCue/7IqWnMY4+e8r+vbEEa32ua/jas//xQ
3//sp0O99b087YB13iKeWub4dq+nvx18wOZQr3mtXRpTtU/0DVc12x5q1lUB4NG1Mdf50gHz05gb
50Yd9fK+S0L90LLD0jJ1W+M2Dm63LY1h/zFr2UDW2nkb52zomJb599eeDHXbJoemMfYFm486/oVr
jDHGlIAfuMYYY0wJ+IFrjDHGlIA1lQ8ASp/9PwecFOorzn0wLXfPlDP+5O9iP+qC+tx/tnvN/9CY
7BPt1zqOGXXsLyt+94a10bvbuvWbaUyrdm+EmnORf/5y1D+B3NuWM5sBYMLSbqFmTXTJ+uhzBbIW
XCMyj1nfPvY/PhPqqw7JmjP7j1WPX9aleRnV83f8bzqEevTBjWmM0reZji12hHrDtma7/feiY25Y
tynU/ZvG3sHTt+fewY07VvoeZf5u8C9cY4wxpgT8wDXGGGNKwA9cY4wxpgSsj/wNaNuyf9BsL2mT
M4/PO+qFUD+y+MiKn9u+RfSAcm9ZAFhJHssVW7M+y/qmyu1lry77Y1mDBID+tctDzXotkDXbTY01
aQzDXlLVZ5c9yqzZKm1z4/b4/1HWioGct9yvdfS+csY0ULk3L5C3ibOrz7t8Ulrm4TtGhnr6upxV
XVsT9WLVr/eSScN3u8zYwXPSMjsa41wA9hEDQIfmWauuBGdTd9wnn4sPbo150Dvfr/d9zXwg8S9c
Y4wxpgT8wDXGGGNKwA9cY4wxpgSsdZTA8OpLgmY79d9/HP79tvFjKn7GcQf9Nv2NtUrWbBvey71j
GaXz8nJqzMItbULNmu2Qgc+kZW6dPjTUyqvJDDj45VCrPOMiPL059oHt2Dxqg0e2zZnCRXytqvfu
riiN/ILDok/4y/MPSmP+ufPu9c4GkSnNOjrrnwDwlR5b6XPyOVJJE1cZ2KxDP70568fs7770yomh
vvrfLkjLsKdaZVOzN5u14nvr16RlnOts/hb4F64xxhhTAn7gGmOMMSXgB64xxhhTAlkIMn8Sn913
XNBnRxz4dhpzaJtXQr3hqdinlvONgayJ9RzyqzRm84NREx1yZfRU/uqHJ6Zl9v/4W6FWmijrb3vi
n2QvJwB0oNhm5T9l7+iC+rivjm+X9y/rqEqPZb9st3a/C3XnjhvSMhe+G/VN1q0BYFjXqDGzX7bh
vazhsof5okO2pzGD+iwK9aIlsf/toftkXZ2Pbc3K7mlMu1ZxuzfR+gJA82ZRW29fFXXfF1/PPmf2
Gx8vpHb2Ot97y7mh/sLh0f8NANc/F4+/6vHL5ycfk9pWeX2rm54Trlt1bK9de5t1XvMXxb9wjTHG
mBLwA9cYY4wpAT9wjTHGmBKwRvEn0GTv1qlv7b+0Gx1qpUOxTrppW1WolZbJ3tfp69qmMRu3vx9q
9okqvyT3LVU9Xov0zB1x4Ovpb7vCWiEAPPbKIaFWPlz23bZoGTW7+XU90jKs4c5+be80prpJ/Bvr
yVeeNDstM2dRn1CfefGUNKb++bhN9z/76VCzNxbIGc3sTwVyD+IN78Z99e0jcg51I2nOymNbRO/m
z7lrZcx+vuef5qZlOEtZHSfWgjknW2U/X9E7+nC5rzGQ86vZozyk24q0zC0LK88NYM2Zz9dfrM/7
99GG231PNf8r/oVrjDHGlIAfuMYYY0wJ+IFrjDHGlID1ht3QpWpo0Gxv6JK9eqyJcU9aBS+jcpJV
ZizDfVV7t18favZ7AkA39mGSngwA7Vs0VhzDvkvOs90TTUx9LqP8sstWdwn1nmwT/zuQ85ePb5d1
a9bW2RPat82WtAzDGikAfO/YZaF+aFn04Q4S++HRtfEYsPcYyHqngvdFlfD8VvrujeJrxvZYF+o5
GzqGWvVQLuIBPr33i6HudEbsu/ydr56XV4ZQx6lSXvjC+qw5920dz/vvvZznSCzeNtn33Y8o/oVr
jDHGlIAfuMYYY0wJ+IFrjDHGlICzlHehbcv+QbP9TIuuNCLrkuyz3CB0ne41UdC6b2PUgqqb5jxb
1tEOPmBzGlPVLOrFnIF7/sipaZnHHj0l1Kqv6gD+LpG3zJot93hVsLd45dsfT2PYm8l6t9JwWUdl
fzIAXNcv6tuszyoWvBX358ZteRn2844dPCfUnIEMAD27rA71Pb2WpjHXzhgcatbEOTcZAPrSvlPw
MVB+ae5/24fW7/z/HJiWmfrvE0LN/mQg74uGnfH/+5wfDQB3zx4U6rN65n117wtHhbpuVv9Q92ud
fc6qTzHDOjnr1NVNcsb45DVxbsCz//pQGlP33MnhPsP9kOdum2CN9+8U/8I1xhhjSsAPXGOMMaYE
/MA1xhhjSuAjrBU0SbnIP+oxJtSc66t6x7IGpryvG7ZHvWhUbdQ7WTMDsnap8mBnUr9TzsVlny6Q
tWGlo/LnKL3wJ7+OWbQdm0edTOUZX/nJjaFmvRbIvsXre2Q9lmGNkfOMgZwPzX7fXn1eSstMfHxI
qKdtzJpdx31iNjXrhcqXzX5k1n2BrJNO/fcfh/qGG7+UluHPKbIflP+Y/afcX3azOGc4t5l7KgPA
1QPnh/rGuVFrVd7tXp3Whnr7jmZpDM9dWLK+cxrDDBv+eKgfmDoijeHrn/OXVZ74CWfHntRf+O7n
0xju6ctZ5Ssasz990e/nhLpxx8qP8L37w4t/4RpjjDEl4AeuMcYYUwIfodcS8RXywBZfSCMuOiRa
JjhWjqPogGwdqK1dlcaMf5ItHvFVlHqtyxG

In [18]:
using GSL

In [19]:
# Clebsches and d-function
function ClebschGordon(j1,m1,j2,m2,j,m)
    factor = (j1+j2-m)%2==1 ? -1 : 1
    factor*sqrt(2*j+1)*sf_coupling_3j(2*j1,2*j2,2*j,2*m1,2*m2,-2*m)
end

function Wignerd(j,m1,m2,cosθ)
#    print("WignerD(",j,",",m1,",",m2,",",cosθ,")")
    (m1 < 0) && print("Error: Something is wrong with Wignerdhat!")
    (m2 == 0) && sqrt(4*π/(2*j+1))*sf_legendre_sphPlm(j,m1,cosθ)
end

function Wignerdhat(j,m1,m2,cosθ)
#    print("WignerD(",j,",",m1,",",m2,",",cosθ,")")
    (m1 < 0) && print("Error: Something is wrong with Wignerdhat!")
    (m2 == 0) && sqrt(4*π/(2*j+1))*sf_legendre_sphPlm(j,m1,cosθ)/sqrt(1-cosθ*cosθ)^(m1)
end


Wignerdhat (generic function with 1 method)

In [20]:
# tensor pre-factors
function Cs(p1,p2,p3,p4)
    s = mnorm(p3+p4)
    p3-p4 - (mnorm(p3)-mnorm(p4))/s*(p3+p4)
end
Bs(p1,p2,p3,p4) = p3+p4
function Ds(p1,p2,p3,p4)
    # minus is removed due to the crossing
    # i is removed to make amplitude real
    [-sum((σ-ν)*(σ-ρ)*(σ-μ)*(ν-ρ)*(ν-μ)*(ρ-μ)*
            p1[σ]*p2[ν]*p3[ρ]
        for σ=1:4, ν=1:4, ρ=1:4)
        for μ=1:4]
end

# kinematical functions
function ζs(j,l,s,zs)
    (l < j-1 || l > j+1) && return 0;
    # three cases
    (l==j) && return 0;
    ps = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    qs = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    (l==j-1) && return (ps*qs)^(j-1)/(4*π)*sqrt((2*j+1)*(2*l+1)/2.0)*
        ClebschGordon(j-1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)
    (l==j+1) && return λ(s,mJpsi^2,mB^2)*(ps*qs)^(j-1)/(4*π)*sqrt((2*j+1)*(2*l+1)/2.0)*
        ClebschGordon(j+1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)
end
function βs(j,l,s,zs)
    (l < j-1 || l > j+1) && return 0;
    # three cases
    (l == j) && return 0;
    ps = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    qs = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    (l == j-1) && return (ps*qs)^(j)/λ(s,mJpsi^2,mB^2)*
        (s+mJpsi^2-mB^2)/sqrt(2)*
        4/(4*π)*sqrt((2*j+1)*(2*l+1)/2.0)*(
            ClebschGordon(j-1,0,1,0,j,0)*Wignerdhat(j,0,0,zs)/sqrt(2)+
            ClebschGordon(j-1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)*zs
        )
    (l == j+1) && return (ps*qs)^(j)*
        (s+mJpsi^2-mB^2)*
        4/(4*π)*sqrt((2*j+1)*(2*j+3)/2.0)*(
            ClebschGordon(j+1,0,1,0,j,0)*Wignerdhat(j,0,0,zs)/sqrt(2)+
            ClebschGordon(j+1,0,1,1,j,1)*Wignerdhat(j,1,0,zs)*zs
        )
end
function δs(j,l,s,zs)
    (l < j-1 || l > j+1) && return 0;
    # three cases
    (l == j-1 || l == j+1) && return 0;
    ps = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    qs = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    (l == j) && return (ps*qs)^(j-1)*sqrt(2.0)/(4*π)*(2*j+1)*Wignerdhat(j,1,0,zs)
end

function Zjl(j,l,p1,p2,p3,p4)
    s = mnorm(p3+p4)
    t = mnorm(p1+p3)
    u = mB^2+mJpsi^2+mπ^2+mK^2-s-t
    zs = (s*(t-u)+(mπ^2-mK^2)*(mJpsi^2-mB^2))/sqrt(λ(s,mB^2,mJpsi^2)*λ(s,mπ^2,mK^2))
#    print("zs = ",zs,"\n")
    Cs(p1,p2,p3,p4)*ζs(j,l,s,zs)+
        Bs(p1,p2,p3,p4)*βs(j,l,s,zs)+
        Ds(p1,p2,p3,p4)*δs(j,l,s,zs)
end

Zjl (generic function with 1 method)

In [21]:
# definition of the amplitude and square of matrix element
# i.e. contracted with lepton current
function ampl(p1,p2,p3,p4,pars)
    amp = [0,0,0,0]
    for j in 1:length(pars)
        ps = pars[j]
        (length(ps) != 3) && continue
        amp += (ps[1] == 0) ? 0 : ps[1]*Zjl(j,j-1,p1,p2,p3,p4);
        amp += (ps[2] == 0) ? 0 : ps[2]*Zjl(j,j,  p1,p2,p3,p4);
        amp += (ps[3] == 0) ? 0 : ps[3]*Zjl(j,j+1,p1,p2,p3,p4);
    end
    amp
end

function amplsq(p1,p2,p3,p4,q1,q2,pars)    
    amp = ampl(p1,p2,p3,p4,pars)
    ampc = conj(amp);
    real(mprod(amp,q1)*mprod(ampc,q2)+mprod(amp,q2)*mprod(ampc,q1)-mprod(amp,ampc)*mJpsi^2/2)
#    real(amp[2:4]⋅ampc[2:4])
end

amplsq (generic function with 1 method)

In [22]:
Zjl(5,6,tree[2,:P1],tree[2,:P2],tree[2,:P3],tree[2,:P4])

4-element Array{Float64,1}:
 173.43  
  42.7272
  31.6246
  38.6442

In [28]:
hists = histogram(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    weights = [amplsq(tree[i,:P1],tree[i,:P2],
    tree[i,:P3],tree[i,:P4],tree[i,:Q1],tree[i,:Q2],
            [[1,0,0]]) for i=1:nEvents],
    nbins=100,norm=true, lab="j=1, l=0"
    )
histogram!(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    weights = [amplsq(tree[i,:P1],tree[i,:P2],
            tree[i,:P3],tree[i,:P4],tree[i,:Q1],tree[i,:Q2],
            [[0,0,1]]) for i=1:nEvents],
    nbins=100,norm=true, lab="j=1, l=2",fillalpha=0.4
    )

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.00 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 j=1, l=0 
 
 
 
 
 j=1, l=2

In [29]:
histogram2d(
    [mnorm(tree[i,:P3]+tree[i,:P4]) for i in 1:nEvents],
    [mnorm(tree[i,:P1]+tree[i,:P3]) for i in 1:nEvents],
    weights = [amplsq(tree[i,:P1],tree[i,:P2],
            tree[i,:P3],tree[i,:P4],tree[i,:Q1],tree[i,:Q2],
    [[0,0,1]]) for i=1:nEvents],
#    weights = [abs(bw(mnorm(tree[i,:P3]+tree[i,:P4]),1.4^2,0.3))^2 for i=1:nEvents],
    nbins=50
    )

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 16 
 
 
 18 
 
 
 20 
 
 
 22 
 
 
<image width="491" height="364" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAesAAAFsCAYAAADsRpHBAAAc/0lEQVR4nO3dy69l+Vne8fe31r6e
a126qrv6grGbFgEbIgExiiIECIQTEBIIgcQAyTBBSMgTxIBhFP6FZJhITDLwIFKCFAklUQQiEHEz
0LZp02513Y+7TtWpU2effV+/lUE5UgZR9Dx2Nv1W9/czfs7S2muvvd+zBu+zSwD/Fzf3/2nv5H/9
+g/J2UHRD/3Pbj2Us31f5Ozdi2M5e7KYyNmIiFenSzn7X0/25ewfrb8mZ59sb8vZZ8uv6hcOwIei
+bBPAAAA/L8xrAEASI5hDQBAcgxrAACSY1gDAJAcwxoAgOQY1gAAJMewBgAgOcoQXnA/sPcrcsPI
77ylH/df/MwfWOdx8pU35ezp2VU5u9yM5OwHs0M5e2Ec93wzlLMREYfDjZy9czmVs9O2ytmJkb1z
qb++Pzx/Imf/vv65nI2ImK3ek7N9v+a7Cx8rPFkDAJAcwxoAgOQY1gAAJMewBgAgOYY1AADJMawB
AEiOYQ0AQHIMawAAkqNYIKHffOULctHJv/rF35ePu3iql4Z87eufkrMREedLvdzjn3z6y3J2OF7L
2X/7335czt6YrOTs1y/25GxExFZ+9yK6Xv8Irjv9uNU4h3O9wyWK8Y0xMR8F/nh+Imc/iNty9vFC
v99qnfGdiJR4sgYAIDmGNQAAyTGsAQBIjmENAEByDGsAAJJjWAMAkBzDGgCA5Ngp/DY0zaG8zfrv
P/Pz8nF/7vNflLOb0yM52y3HcvZLf/oDcjYi4vb5NTl7PNZ3nN8+uyJnXzZ2p09XIzl7MDAWnCPi
y+f6dR4b/y6f6SvnMTA+2U83+lL2fqsf+PZyqZ9ERNwc6tft/mYuZ+ehn8cHzT05+3D2P+RsHxu+
a/Ft4ckaAIDkGNYAACTHsAYAIDmGNQAAyTGsAQBIjmENAEByDGsAAJJjWAMAkByL+t+GX7n2BblN
4nd/8k/k417O9uXsgycvydn1diBn780O5WxExKCpcrbv9dtuVfX/Jy+3rZz96vlQzo7Mf2mvjPSS
kXtz/VqcLPVWlGGjn3TVT9di9KdERMRF3cjZWSzkbDG+5mbNTM5ujLKVk/VX5WxExHJzImf7fsn3
+McAT9YAACTHsAYAIDmGNQAAyTGsAQBIjmENAEByDGsAAJJjWAMAkBzDGgCA5Fim/z98dvqrVj3E
F3/2bTn7pXffkrNXJnrhw/G+XuIwX07k7N8+ekXORkRsjfKSO/ORnK1GgUpb9LfvPf2yxfHQ+5j8
xeWZnP2xK1fl7L25/vomRiPJ2bqTs+uql9+sez0bEfG4PJOz+/1Uzp6Vczm7LHM5W0MvcanhNc9c
1sdy9nzxNf08eqOFB6nwZA0AQHIMawAAkmNYAwCQHMMaAIDkGNYAACTHsAYAIDmGNQAAyX3kd+6a
MpUXHP/LD/+0dexVN5Cz0+Fazr59elPONsZu8arT/zfbG+i7txERHyz13emTRStnn+qXLa6P9ezF
1shuvB3Zeadfu1sT/R56sNRP+lldydkLY7e4N/aFV0U/h4iIw3ooZ5dlKWcf9u/K2avNq3J2Ffp1
W1R9hzwioin6Z2RdL+XsqtPPY7l+IGcjth/5WfJh48kaAIDkGNYAACTHsAYAIDmGNQAAyTGsAQBI
jmENAEByDGsAAJJjWAMAkJzeyPCC+pef+DU5Ox7esY596/qpnH373nfI2TePn8rZP7j/spx9bbqR
s/OtXsoQEbGpeifCxDj0ZVfl7EzvhohV1Y87bLy+h8uql5e8P9fPY1D0/603vX7c3inWMcpIaujn
EBGxKXoDzmnck7OT5kDOntX7crb2evnNqNmXsxERq3ohZ52iGsdwcE3Obrtz+ST6fkWByreAJ2sA
AJJjWAMAkBzDGgCA5BjWAAAkx7AGACA5hjUAAMkxrAEASI5hDQBAci9kKUpT9uQF/B99TS85+J8P
X7fO4+F7Qzn7xt5Kzj4+O5azT/TDxoO5/nYfDLzegutjvZjhb8/1cpaXRvo1fn+lt6IclrGcPemM
tpWIuF725OxZv5CzW6OE47zVi3UO65F+3KJfi3nVzyEiYlg+IWedIhCnnKUtIzk7KPpn5HL7SM5G
REyNQpL5Rj92E3ojUV/074tBe0XObrdPrBaXPjaUqARP1gAApMewBgAgOYY1AADJMawBAEiOYQ0A
QHIMawAAkmNYAwCQ3Au5Z/3P9z8vZwftXTl7ONha53F0pO8LO4uF95/qO8A3J/qRHy31dcWztb6b
GhFxstTPY9zo/yOerfX3xFnGvBencnYvpsaRI94t9+RsX/Tr3Bgf10k/kbOnzYmcdRw0170/MD4k
tdfvi7bou/rV2GVfV33n3DnfiIjLzQdytqt62UIxdsOd4zbGNW4a/d6MiKh9K98ZfW98yb1geLIG
ACA5hjUAAMkxrAEASI5hDQBAcgxrAACSY1gDAJAcwxoAgOQY1gAAJPdClqJ87pZeJPF3pzfl7Kp6
/7vcudR/qP7hQt/VHxin8ddP9bKFlVH4MDF+eD4iYmmUPoyilbPz0ItnGuN/T6c05Fm5kLMREcU4
j3k8k7MHcVXOPi4P5WwN/fM0Cv26XVS9eCYiYtvoJRzrTi8kWVXv/dPpLS5N0e/5iIja6/e9U3Ti
HNfhlb5437NO4UrtO71AJTYvVIEKT9YAACTHsAYAIDmGNQAAyTGsAQBIjmENAEByDGsAAJJjWAMA
kBzDGgCA5NKUolyZfr+8zP657/6KfNz/+JXPyNlF5/3v8vqeXjBwb64XqNydL+WsXssQ8dpYL7Q4
2zglBxGXvV5o8Q2zZEQ17adydmDc+k42IuK0vydnnbKMRejXbV0XOzmHi/pIzg7KWM5+80zk5NZ4
faXox3UKbZwikBp6IVFERO8c2yo60XtAnOtW61rO9kYJT0REsc7Z+Kz2vVGgsv3QC1R4sgYAIDmG
NQAAyTGsAQBIjmENAEByDGsAAJJjWAMAkBzDGgCA5NLsWf/E6Efl7F/e1vdN58bu9IO597/Luxf6
7vSzrb5nedjqP7Y+6/R9zHeXz+Ss84P2ERFd6OcxCP31rYu+v3lZLuVs0xu7xeWJnI2I6Iwd2XHs
y9nLeiZn26Jf497Y1nf2m2tx9n8j1v1czg5b/bptq95bsLH2043vi97bLe6q3lvg7BbXXv88eeds
ZHtz59xqkzAY/QLeSexmJ5snawAAkmNYAwCQHMMaAIDkGNYAACTHsAYAIDmGNQAAyTGsAQBIjmEN
AEByaUpRro70PfK26Av4f2X0WVwf69mIiIdLvfShNUpGOqOM4OvNe3J2FBM569YQzOOpnD3ub8jZ
RegFOMteL30ZFb1UY5e/Oj+rp3K2LfoNuqozObvu9Gs8bo/kbNd7pSjVKAJxSkN6o7CjbfRrvO2M
gpheP98Is4PDKOEpxt3sfAfsrLgkIpxPoPNeO69wV9fNwZM1AADJMawBAEiOYQ0AQHIMawAAkmNY
AwCQHMMaAIDkGNYAACTHsAYAILmdlqK0zbG8Hz5p9eP+3nuHcnZ/oC+z/9XcaFCJiEmM5OzCKEWY
N3PjHPRyj7P+gZwdFL1AJcIrRZgVvUBlY1y3ZaeXokSr/5+6rXr5RURENYoZhkbRyaJ7LGeL8dF2
ik62/VLPGqUhERGjgX4e1SgCcXR1LWf76HZyDhFeCYd

## Analytical approach
Now, I get the same distributions analytically
 - Matrix element square is lorethz invariant
 - in s-channel CMS, amplitude is straightforward
 - integrals over phase space are analytical as long as amplitude depend on (s, sz)
 - avaraging over J/psi polarization is a little bit tricky (see comment in the cell below)

In [30]:
function amplsp_analytic(s,zs,pars)
    # states along z axis
    p = sqrt(λ(s,mB^2,mJpsi^2)/(4*s))
    E1 = sqrt(p^2+mJpsi^2)
    E2 = sqrt(p^2+mB^2)
    p1 = [E1, 0,0,p]
    p2 = [E2, 0,0,p]
    # scattered states
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    E3 = sqrt(q^2+mπ^2)
    E4 = sqrt(q^2+mK^2)
    sinθ = sqrt(1-zs^2)
    p3 = [E3,-q*sinθ,0,-q*zs]
    p4 = [E4, q*sinθ,0,q*zs]
    # amplitude
    amp = ampl(p1,p2,p3,p4,pars)
    ampc = conj(amp)
    # avaraging over muon angles is easy in J/psi rest frame
    # the answer there is \vec A \cdot \vec A (some numerical factor)
    # therefore, I have to find \vec A^2 in Jpsi rest frame
    # I do it using scalar product of A to p1
    vecTildeAsq = mprod(amp,p1)*mprod(p1,ampc)/mJpsi^2 - mprod(amp,ampc)
    # attach muon current avaraged over muons angles
    ksq = λ(mJpsi,mμ^2,mμ^2)/(4*mJpsi^2)
    return 2*vecTildeAsq*(mJpsi^2/4-ksq/3)
end

amplsp_analytic (generic function with 1 method)

In [31]:
plot(0.5:0.03:2.5, -.99:0.03:0.99, 
    (s,zs)->(sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s*
amplsp_analytic(s,zs,[[0,0,1]])), t=:heatmap)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 -1.0 
 
 
 -0.5 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
<image width="475" height="364" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAdsAAAFsCAYAAACEtRP5AAAgAElEQVR4nO3dy3Jk2ZFeYT+XCACZ
WUU2KZGSKGuapJkeTU/SL9qSWZs11azMBBC3ExrANBJ97aIDHgWU1jf12ieugONY/uU+/dN//R/X
AP/9t/+W1v7bH/9XWvsPf/7ntPb5v+Tn1j9PaW37039Oa6c//Dk/95t/TGvXL39Ka/v979PasnxO
axER07zmxWv+lm/bMa1dLt/T2vn0U/5wh39Na9PTv6S1+ftf0tryLa/NP0Htr/n3KSIi/vo1LW1/
zY9dvt6ntfNPn9La8af8c3z+mteevufXfHp6SGuPz/nz/Ea1411ai4j4dtrnj3na5efOS1r7jrUZ
avnP8OMlLcXTGWqX/Gfmacsv+rTBRSPiKU5p7TkOeW16TmuH6SmtHSOvna75Nc9Qu1zhdwb8Ptnw
XP6+vJzN39drwHt+zT+ra2xwLq/hObwmtj54vNq5/CdGkiS9CZutJEnNbLaSJDWz2UqS1MxmK0lS
s/UJUoUREd8hyUjJyqefvqS1/V9+SGvzPSRZ5/+Z1nbnPAF3eYKE65c8GX26/zGvrXkaNSLiOufv
W0CabbrkCcE450nG6ZgnlZdD/vrnR6h9y+O/07c8/Tx9/ZbWrl8HKcev+ffx8o0Sx3ly+Pgt/6wO
UHt6zFPFT0/5c3l8zpPD3w957bGYKI6IeD7n6fenS/6ePhdrhy1PHB8gAHqENPJpy38uTpBGpXNn
TKpGnCN/QucJfqdAbYM0LtWulNSlNG4xqXvFNO4ocVtL5Nav2fF45O0fzztbSZKa2WwlSWpms5Uk
qZnNVpKkZuvTJQ86REQ8XfLgxbfnPEDyHUbd3f3vPHi03OUBoXXOgzfzOQ86TU95eOj6JR9leL3P
X8O25/F5Meej7sh0yQMU0yl/b6YjhKeeofaYvzfxmJ+7focxcPnHFNtjHgKKiDh/y79TVDt+z2sH
qD1TCIoCgAcISFEICsYuUkDqGX4OIyKeLvn3jWrlEBQEnY6QSTpB7uQEgR0OQeVP5gS1l7MQgoKz
FzwHASkcV1gbZdgx5nA8khDqrwpeVdw6PFXjna0kSc1stpIkNbPZSpLUzGYrSVIzm60kSc3W50Ea
+TuMgaOF1o+wQPv+r/kox2Wfj/O7n/PU2Xp+TGvTU544nj/n4wrjPn99847H58Vc/DsGFmFPJ1jK
fIQxiIc8OXx9hkRi/pbG9py//u0xf9/O3/MUb0TE6TGvnyA5fIBzzzBa8RmSw0/VVDHUcPwpjlzk
dHs1cfwMieNnCPLSSMYT1vKf4WN1XCONXBykkWnsYn0kYy1VvBUTxzyusZoMfs24xryGV/0YoeIy
72wlSWpms5UkqZnNVpKkZjZbSZKarRSCiIh4POfBi2/HfLzc16d8T+jdVxjJuINxhfQP7/A8l8Nz
WpvhuUwwOjIG+ajynzE0eQ3yUVfIR12P+ZO5QgiIQlAX2Nl6fsprJ6hFRBwhzHSEs88wPvFwyL+n
TxBmeoLvd3X37NMZahByop/Dl7MUdKrVqntpKejUEoKCc7STNoIDVPWRjMVxjR07a4v7bHnX7cuV
a97Tztrb8s5WkqRmNltJkprZbCVJamazlSSpmc1WkqRmK0zri4iIRxj19gjJym+QDn2AUY7LT4N4
dGKDtOYOUqzLwyGtzTA6cloHbxyMlkQwPu8KnwWlsbdjPgZwO+af4RlSvOdnqMG5I6SYIyKOcPYA
6WCqPVPiuCNVXBy7+ASf4RN89hGDkYyUOKZUcTFxTCMZMXFMqWJI1Z4gcTxKI+O4xsh//mkkI49r
fPvl8bSsnVPF9DuMf3+9ploxXmb//nlnK0lSM5utJEnNbLaSJDWz2UqS1Gw9Xvgfnh/PebiCdt1+
hcDKPYxyXGYKAuSlC4RL9jCScHcPASkYHTmvHOSapto/6F+vtYAUBcQ2+JwuEJ46Q3jqBJ/vCc4d
4VxExBFCSQcIMz3DOapx0AmuWQ461UJQ44AUjGSEHykcyVjcWcvhqdpIRt5ZCwEpqEVwCArHNcJI
xurYxa1hXCOFoOq7bn9O/a3PfXze2UqS1MxmK0lSM5utJEnNbLaSJDWz2UqS1Gw9QHIwglOQ3yB1
+RkSoHcwyhHTyOACKc87SMDuYFn5skIaeRmkkavjGiGNvG2URs5rF0ojw2d4gs/wDNc8QY3SxhER
B0oVQzr4QMlheD7PF6rl7w2mkfFc/vnSzxqde6nntY6RjMfi2MVjOXGcp4ZpJCONY4wYLIGHGo1W
rI9yrC6B71gQP/j99Zqzb37uY/DOVpKkZjZbSZKa2WwlSWpms5Ukqdl6gPFpERFPELx5hNp3CKzs
YdYbBaTo3+TPEEo5wXPZrXnwYoEQ1Lzw+zZPtaAXjWvcrhCQgnDNZcvfGwqWnSk8RSEoCB0d4VxE
xAHqBwoz0Tl4/YdiCIr3x+a1A+wrxpGLgzXPVKexixiCgt8NHJ6iEBTtrIVzxZ21NI7xpU4jGSHo
BLUr1cpBp46xixCeekVYCU+2ZKDgohjkui3vbCVJamazlSSpmc1WkqRmNltJkpoNA1K0C/M7BDpo
utRuzgNLM+yB3SA8dIIQzP05D0nslzwgsVJAajDpqrrPluAEKXhvKCBFwTKqHSmQRgEpOBfBgaUj
vA667gHeNwpBHeD7zdeEc9W9s4OAFIagYGc1B6RgLy2GoGq7Z080JYqCTMWdtKN6dRLUVgxB0e7Z
ao133b7md9Tbh7Lqj/cxeGcrSVIzm60kSc1stpIkNbPZSpLUzGYrSVKz9TBI6x0ueXL46Qxp5CXv
47s5T6tSGplGGZ4hHUojAPeQOKY08jJIG1MamTaT0lUpcUyjHC/w3tD7RrVTMTV8gmuO6tXE8ZG+
G8XEMaWDaSQj7palNPJg+ucJEse0s5r20mKNRivSrtvi2MXzRInjWi0iYrvCzlqs0UjG4ijHhr20
5Z21v8CYw9eMiPwIvLOVJKmZzVaSpGY2W0mSmtlsJUlqth4hBBAR8XzJ/9H+EQIrdxCe2k15j58i
DzPxuMb8mndLfs3dDAEpGMk4CkhR0ItgnAEDUjCuEcJTZwjznPFcMVg1CEgd4flQ0IlrdM3aaMVq
mOkIASlY84yjE0d1DDrBuNYTjlak3bN5KIlq1RAUj3Lk32/1kYzFoFNxtGK11jdWsWMk46+bd7aS
JDWz2UqS1MxmK0lSM5utJEnNbLaSJDVbD9MR/4Pn7T6vXfJe/Qhp5BXSyDTLkNPItRTrbs4T1ZQ4
Ho5rpERecV7jFQ5yGrl27ozjMeGzgBQzfU4v9dpZShVzyreWOKYaPx6dqy1yfzlbHbuYp3VxJCOk
gzFxDI93Li6Bp6TyBuMRX+qwPL6aKi6PZLxt4pi/UaMq/qIqX/e96HiW3tlKktTMZitJUjObrSRJ
zWy2kiQ1W4+jgBQECJ5hXOMeg0f0iHn/x4AU1Gif5w5GMq4QghqNY8S/YugsvA7edUu1WngKA1LF
8BSdi+AQFAWP6PPHcw0hKAo6nenxaA/saFwjhHJotCKHoGBcI4agaCRj/vumupeWQ071cY3lvbQd
Yxc/1M7ajxGCuvXz9M5WkqRmNltJkprZbCVJamazlSSpmc1WkqRm63E64H9wgDTf02WX1nZz3sdn
nFeYqy6P38+URs6vSanp8bhGqEGRAoLVdc2cR

In [27]:
# S-wave
function projected_dalitz(s,par)
    sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s*
        quadgk(zs->amplsp_analytic(s,zs,par),
                -0.999,0.999)[1]
end

plot(hists)
norm1 = quadgk(s->projected_dalitz(s,[[1,0,0]]),((mπ+mK)^2+0.01),((mB-mJpsi)^2-0.001))[1]
plot!(s->projected_dalitz(s,[[1,0,0]])/norm1, 
    ((mπ+mK)^2+0.01):0.02:((mB-mJpsi)^2-0.001),
    lab="j=1, l=0")
# D-wave
norm2 = quadgk(s->projected_dalitz(s,[[0,0,0.01]]),((mπ+mK)^2+0.01),((mB-mJpsi)^2-0.001))[1]
plot!(s->projected_dalitz(s,[[0,0,0.01]])/norm2,
    ((mπ+mK)^2+0.01):0.02:((mB-mJpsi)^2-0.001),
    lab="j=1, l=2")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.00 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
<polyline clip-path="url(#clip8002)" style="stroke:#3da44d; stroke-width:1; stroke-opacity:1; fill:none" points="
 58.7198,374.866 63.4499,371.089 68.1801,366.056 72.9102,360.294 77.6403,354.081 82.3704,347.593 87.1005,340.95 91.8306,334.239 96.5607,327.523 101.291,320.846 
 106.021,314.244 110.751,307.742 115.481,301.357 120.211,295.105 124.941,288.994 129.672,283.031 134.402,277.221 139.132,271.565 143.862,266.066 148.592,260.724 
 153.322,255.537 158.052,250.506 162.782,245.627 167.512,240.899 172.243,236.32 176.973,231.888 181.703,227.599 186.433,223.451 191.163,219.442 195.893,215.57 
 200.623,211.831 205.353,208.223 210.083,204.744 214.814,201.391 219.544,198.162 224.274,195.056 229.004,192.07 233.734,189.202 238.464,186.45 243.194,183.813 
 247.924,181.289 252.654,178.876 257.385,176.573 262.115,174.38 266.845,172.294 271.575,170.314 276.305,168.44 281.035,166.671 285.765,165.006 290.495,163.443 
 295.225,161.984 299.956,160.626 304.686,159.371 309.416,158.217 314.146,157.164 318.876,156.212 323.606,155.361 328.336,154.612 333.066,153.964 337.796,153.419 
 342.527,152.975 347.257,152.634 351.987,152.397 356.717,152.264 361.447,152.236 366.177,152.314 370.907,152.5 375.637,152.795 380.367,153.199 385.098,153.715 
 389.828,154.345 394.558,155.091 399.288,155.954 404.018,156.937 408.748,158.044 413.478,159.275 418.208,160.636 422.938,162.129 427.669,163.758 432.399,165.527 
 437.129,167.441 441.859,169.505 446.589,171.724 451.319,174.105 456.049,176.655 460.779,179.381 465.509,182.291 470.24,185.396 474.97,188.706 479.7,192.233 
 484.43,195.991 489.16,199.996 493.89,204.267 498.62,208.826 503.35,213.697 508.08,218.912 512.811,224.507 517.541,230.528 522.271,237.032 527.001,244.091 
 531.731,251.802 536.461,260.298 541.191,269.769 545.921,280.507 550.651,293.007 555.382,308.242 560.112,328.804 
 "/>
<polyline clip-path="url(#clip8002)" style="stroke:#c271d2; stroke-width:1; stroke-opacity:1; fill:none" points="
 58.7198,371.16 63.4499,353.202 68.1801,330.326 72.9102,305.354 77.6403,279.737 82.3704,254.348 87.1005,229.74 91.8306,206.271 96.5607,184.165 101.291,163.557 
 106.021,144.52 110.751,127.079 115.481,111.231 120.211,96.9475 124.941,84.1866 129.672,72.8942 134.402,63.0096 139.132,54.4676 143.862,47.2005 148.592,41.1396 
 153.322,36.2166 158.052,32.3639 162.782,29.5155 167.512,27.6073 172.243,26.5774 176.973,26.3662 181.703,26.9165 186.433,28.1739 191.163,30.0863 195.893,32.604 
 200.623,35.6798 205.353,39.269 210.083,43.3292 214.814,47.8199 219.544,52.703 224.274,57.9425 229.004,63.5041 233.734,69.3556 238.464,75.4663 243.194,81.8074 
 247.924,88.3516 252.654,95.0732 257.385,101.948 262.115,108.952 266.845,116.066 271.575,123.267 276.305,130.537 281.035,137.857 285.765,145.211 290.495,152.582 
 295.225,159.956 299.956,167.318 304.686,174.654 309.416,181.952 314.146,189.201 318.876,196.389 323.606,203.505 328.336,210.541 333.066,217.486 337.796,224.333

## Example of the interest
Consider two resonances:
 * $K^*(1410)$, $m = 1.414\,$MeV, $\Gamma = 232\,$MeV
 * $K_2^*(1430)$, $m = 1.432\,$MeV, $\Gamma = 109\,$MeV

In [32]:
mKs = 1.414; ΓKs = 0.232;
mKs2 = 1.432; ΓKs2 = 0.109;
function bwDalitz(s,msq,Γ)
    phsp = sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s
    phsp*abs(bw(s,msq,Γ))^2
end
plot(s->bwDalitz(s,mKs^2,ΓKs), (mπ+mK)^2:0.01:(mB-mJpsi)^2, lab="K*(1410)")
plot!(s->bwDalitz(s,mKs2^2,ΓKs2), (mπ+mK)^2:0.01:(mB-mJpsi)^2, lab="K2*(1430)", legend=:topleft)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0 
 
 
 50 
 
 
 100 
 
 
 150 
 
 
 200 
 
<polyline clip-path="url(#clip8802)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.5169,375.813 38.0627,374.752 40.6085,374.309 43.1543,373.966 45.7001,373.675 48.2458,373.416 50.7916,373.18 53.3374,372.961 55.8832,372.755 58.429,372.559 
 60.9748,372.372 63.5206,372.192 66.0663,372.018 68.6121,371.849 71.1579,371.685 73.7037,371.524 76.2495,371.366 78.7953,371.21 81.3411,371.057 83.8868,370.905 
 86.4326,370.755 88.9784,370.607 91.5242,370.459 94.07,370.312 96.6158,370.166 99.1616,370.02 101.707,369.874 104.253,369.728 106.799,369.582 109.345,369.436 
 111.89,369.289 114.436,369.142 116.982,368.994 119.528,368.845 122.074,368.695 124.619,368.544 127.165,368.392 129.711,368.238 132.257,368.083 134.803,367.927 
 137.348,367.768 139.894,367.608 142.44,367.447 144.986,367.283 147.531,367.117 150.077,366.948 152.623,366.778 155.169,366.605 157.715,366.429 160.26,366.251 
 162.806,366.07 165.352,365.886 167.898,365.699 170.444,365.509 172.989,365.316 175.535,365.119 178.081,364.918 180.627,364.714 183.172,364.506 185.718,364.294 
 188.264,364.078 190.81,363.858 193.356,363.633 195.901,363.404 198.447,363.169 200.993,362.93 203.539,362.686 206.085,362.436 208.63,362.181 211.176,361.92 
 213.722,361.653 216.268,361.38 218.813,361.101 221.359,360.815 223.905,360.523 226.451,360.223 228.997,359.916 231.542,359.601 234.088,359.279 236.634,358.948 
 239.18,358.609 241.726,358.262 244.271,357.905 246.817,357.539 249.363,357.163 251.909,356.777 254.454,356.38 257,355.973 259.546,355.554 262.092,355.124 
 264.638,354.682 267.183,354.227 269.729,353.76 272.275,353.279 274.821,352.783 277.367,352.274 279.912,351.749 282.458,351.209 285.004,350.653 287.55,350.08 
 290.095,349.49 292.641,348.882 295.187,348.256 297.733,347.61 300.279,346.944 302.824,346.258 305.37,345.55 307.916,344.821 310.462,344.068 313.008,343.292 
 315.553,342.492 318.099,341.667 320.645,340.816 323.191,339.938 325.736,339.033 328.282,338.1 330.828,337.138 333.374,336.147 335.92,335.125 338.465,334.072 
 341.011,332.988 343.557,331.873 346.103,330.724 348.649,329.544 351.194,328.33 353.74,327.084 356.286,325.805 358.832,324.493 361.377,323.15 363.923,321.776 
 366.469,320.372 369.015,318.939 371.561,317.48 374.106,315.996 376.652,314.49 379.198,312.965 381.744,311.425 384.289,309.874 386.835,308.317 389.381,306.759 
 391.927,305.207 394.473,303.667 397.018,302.146 399.564,300.654 402.11,299.199 404.656,297.79 407.202,296.437 409.747,295.152 412.293,293.944 414.839,292.826 
 417.385,291.809 419.93,290.903 422.476,290.12 425.022,289.47 427.568,288.962 430.114,288.606 432.659,288.408 435.205,288.375 437.751,288.511 440.297,288.819 
 442.843,289.3 445.388,289.952 447.934,290.773 450.48,291.757 453.026,292.899 455.571,294.19 458.117,295.62 460.663,297.179 463.209,298.854 465.755,300.633 
 468.3,302.504 470.846,304.452 473.392,306.465 475.938,308.53 478.484,310.635 481.029,312.766 483.575,314.914 486.121,317.068 488.667,319.218 491.212,321.356 
 493.758,323.474 496.304,325.565 498.85,327.625 501.396,329.647 503.941,331.628 506.487,333.564 509.033,335.453 511.579,337.294 514.125,339.083 516.67,340.822 
 519.216,342.509 521.762,344.145 524.308,345.73 526.853,347.265 529.399,348.751 531.945,350.188 534.491,351.58 537.037,352.928 539.582,354.233 542.128,355.497 
 544.674,356.724 547.22,357.915 549.766,359.073 552.311,360.202 554.857,361.304 557.403,362.384 559.949,363.445 562.494,364.494 565.04,365.537 567.586,366.582 
 570.132,367.643 572.678,368.737 575.223,369.896 577.769,371.184 580.315,372.781 
 "/>
<polyline clip-path="url(#clip8802)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 35.5169,375.813 38.0627,374.784 40.6085,374.354 43.1543,374.022 45.7001,373.739 48.2458,373.488 50.7916,373.259 53.3374,3

In [33]:
plot(0.5:0.03:2.5, -.99:0.03:0.99,
    (s,zs)->(sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s*
amplsp_analytic(s,zs,[[0,0,1]])), t=:heatmap)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 -1.0 
 
 
 -0.5 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
<image width="475" height="364" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAdsAAAFsCAYAAACEtRP5AAAgAElEQVR4nO3dy3Jk2ZFeYT+XCACZ
WUU2KZGSKGuapJkeTU/SL9qSWZs11azMBBC3ExrANBJ97aIDHgWU1jf12ieugONY/uU+/dN//R/X
AP/9t/+W1v7bH/9XWvsPf/7ntPb5v+Tn1j9PaW37039Oa6c//Dk/95t/TGvXL39Ka/v979PasnxO
axER07zmxWv+lm/bMa1dLt/T2vn0U/5wh39Na9PTv6S1+ftf0tryLa/NP0Htr/n3KSIi/vo1LW1/
zY9dvt6ntfNPn9La8af8c3z+mteevufXfHp6SGuPz/nz/Ea1411ai4j4dtrnj3na5efOS1r7jrUZ
avnP8OMlLcXTGWqX/Gfmacsv+rTBRSPiKU5p7TkOeW16TmuH6SmtHSOvna75Nc9Qu1zhdwb8Ptnw
XP6+vJzN39drwHt+zT+ra2xwLq/hObwmtj54vNq5/CdGkiS9CZutJEnNbLaSJDWz2UqS1MxmK0lS
s/UJUoUREd8hyUjJyqefvqS1/V9+SGvzPSRZ5/+Z1nbnPAF3eYKE65c8GX26/zGvrXkaNSLiOufv
W0CabbrkCcE450nG6ZgnlZdD/vrnR6h9y+O/07c8/Tx9/ZbWrl8HKcev+ffx8o0Sx3ly+Pgt/6wO
UHt6zFPFT0/5c3l8zpPD3w957bGYKI6IeD7n6fenS/6ePhdrhy1PHB8gAHqENPJpy38uTpBGpXNn
TKpGnCN/QucJfqdAbYM0LtWulNSlNG4xqXvFNO4ocVtL5Nav2fF45O0fzztbSZKa2WwlSWpms5Uk
qZnNVpKkZuvTJQ86REQ8XfLgxbfnPEDyHUbd3f3vPHi03OUBoXXOgzfzOQ86TU95eOj6JR9leL3P
X8O25/F5Meej7sh0yQMU0yl/b6YjhKeeofaYvzfxmJ+7focxcPnHFNtjHgKKiDh/y79TVDt+z2sH
qD1TCIoCgAcISFEICsYuUkDqGX4OIyKeLvn3jWrlEBQEnY6QSTpB7uQEgR0OQeVP5gS1l7MQgoKz
FzwHASkcV1gbZdgx5nA8khDqrwpeVdw6PFXjna0kSc1stpIkNbPZSpLUzGYrSVIzm60kSc3W50Ea
+TuMgaOF1o+wQPv+r/kox2Wfj/O7n/PU2Xp+TGvTU544nj/n4wrjPn99847H58Vc/DsGFmFPJ1jK
fIQxiIc8OXx9hkRi/pbG9py//u0xf9/O3/MUb0TE6TGvnyA5fIBzzzBa8RmSw0/VVDHUcPwpjlzk
dHs1cfwMieNnCPLSSMYT1vKf4WN1XCONXBykkWnsYn0kYy1VvBUTxzyusZoMfs24xryGV/0YoeIy
72wlSWpms5UkqZnNVpKkZjZbSZKarRSCiIh4POfBi2/HfLzc16d8T+jdVxjJuINxhfQP7/A8l8Nz
WpvhuUwwOjIG+ajynzE0eQ3yUVfIR12P+ZO5QgiIQlAX2Nl6fsprJ6hFRBwhzHSEs88wPvFwyL+n
TxBmeoLvd3X37NMZahByop/Dl7MUdKrVqntpKejUEoKCc7STNoIDVPWRjMVxjR07a4v7bHnX7cuV
a97Tztrb8s5WkqRmNltJkprZbCVJamazlSSpmc1WkqRmK0zri4iIRxj19gjJym+QDn2AUY7LT4N4
dGKDtOYOUqzLwyGtzTA6cloHbxyMlkQwPu8KnwWlsbdjPgZwO+af4RlSvOdnqMG5I6SYIyKOcPYA
6WCqPVPiuCNVXBy7+ASf4RN89hGDkYyUOKZUcTFxTCMZMXFMqWJI1Z4gcTxKI+O4xsh//mkkI49r
fPvl8bSsnVPF9DuMf3+9ploxXmb//nlnK0lSM5utJEnNbLaSJDWz2UqS1Gw9Xvgfnh/PebiCdt1+
hcDKPYxyXGYKAuSlC4RL9jCScHcPASkYHTmvHOSapto/6F+vtYAUBcQ2+JwuEJ46Q3jqBJ/vCc4d
4VxExBFCSQcIMz3DOapx0AmuWQ461UJQ44AUjGSEHykcyVjcWcvhqdpIRt5ZCwEpqEVwCArHNcJI
xurYxa1hXCOFoOq7bn9O/a3PfXze2UqS1MxmK0lSM5utJEnNbLaSJDWz2UqS1Gw9QHIwglOQ3yB1
+RkSoHcwyhHTyOACKc87SMDuYFn5skIaeRmkkavjGiGNvG2URs5rF0ojw2d4gs/wDNc8QY3SxhER
B0oVQzr4QMlheD7PF6rl7w2mkfFc/vnSzxqde6nntY6RjMfi2MVjOXGcp4ZpJCONY4wYLIGHGo1W
rI9yrC6B71gQP/j99Zqzb37uY/DOVpKkZjZbSZKa2WwlSWpms5Ukqdl6gPFpERFPELx5hNp3CKzs
YdYbBaTo3+TPEEo5wXPZrXnwYoEQ1Lzw+zZPtaAXjWvcrhCQgnDNZcvfGwqWnSk8RSEoCB0d4VxE
xAHqBwoz0Tl4/YdiCIr3x+a1A+wrxpGLgzXPVKexixiCgt8NHJ6iEBTtrIVzxZ21NI7xpU4jGSHo
BLUr1cpBp46xixCeekVYCU+2ZKDgohjkui3vbCVJamazlSSpmc1WkqRmNltJkpoNA1K0C/M7BDpo
utRuzgNLM+yB3SA8dIIQzP05D0nslzwgsVJAajDpqrrPluAEKXhvKCBFwTKqHSmQRgEpOBfBgaUj
vA667gHeNwpBHeD7zdeEc9W9s4OAFIagYGc1B6RgLy2GoGq7Z080JYqCTMWdtKN6dRLUVgxB0e7Z
ao133b7md9Tbh7Lqj/cxeGcrSVIzm60kSc1stpIkNbPZSpLUzGYrSVKz9TBI6x0ueXL46Qxp5CXv
47s5T6tSGplGGZ4hHUojAPeQOKY08jJIG1MamTaT0lUpcUyjHC/w3tD7RrVTMTV8gmuO6tXE8ZG+
G8XEMaWDaSQj7palNPJg+ucJEse0s5r20mKNRivSrtvi2MXzRInjWi0iYrvCzlqs0UjG4ijHhr20
5Z21v8CYw9eMiPwIvLOVJKmZzVaSpGY2W0mSmtlsJUlqth4hBBAR8XzJ/9H+EQIrdxCe2k15j58i
DzPxuMb8mndLfs3dDAEpGMk4CkhR0ItgnAEDUjCuEcJTZwjznPFcMVg1CEgd4flQ0IlrdM3aaMVq
mOkIASlY84yjE0d1DDrBuNYTjlak3bN5KIlq1RAUj3Lk32/1kYzFoFNxtGK11jdWsWMk46+bd7aS
JDWz2UqS1MxmK0lSM5utJEnNbLaSJDVbD9MR/4Pn7T6vXfJe/Qhp5BXSyDTLkNPItRTrbs4T1ZQ4
Ho5rpERecV7jFQ5yGrl27ozjMeGzgBQzfU4v9dpZShVzyreWOKYaPx6dqy1yfzlbHbuYp3VxJCOk
gzFxDI93Li6Bp6TyBuMRX+qwPL6aKi6PZLxt4pi/UaMq/qIqX/e96HiW3tlKktTMZitJUjObrSRJ
zWy2kiQ1W4+jgBQECJ5hXOMeg0f0iHn/x4AU1Gif5w5GMq4QghqNY8S/YugsvA7edUu1WngKA1LF
8BSdi+AQFAWP6PPHcw0hKAo6nenxaA/saFwjhHJotCKHoGBcI4agaCRj/vumupeWQ071cY3lvbQd
Yxc/1M7ajxGCuvXz9M5WkqRmNltJkprZbCVJamazlSSpmc1WkqRm63E64H9wgDTf02WX1nZz3sdn
nFeYqy6P38+URs6vSanp8bhGqEGRAoLVdc2cR

## Compare three amplitudes
* Our amplitude, Eq. (B1)
* LS-amplitude, Eq. (13)
* Tensor amplitude, Eq. (19)

Partial wave expansion for the amplitude
$$
A_\lambda(s,\cos\theta) = \frac{1}{4\pi}\sum (2j+1) A_\lambda^j(s) d_{\lambda 0}^j(\theta)
$$

Integrating over muons is equivalent to summing over helicity
$$
\frac{\textrm{d}I}{\textrm{d}s} = N\big( |A_0^1(s)|^2 + 2|A_1(s)|^2 \big) \frac{\lambda_{\pi K}^{1/2}(s)
\lambda_{\psi B}^{1/2}(s)}{s}
$$
Below, I demonstrate it numerically.

In [34]:
# our amplitude
function A10_our(s,cS,cD)
    j = 1
    lam=0
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    K00 = 2*mJpsi/sqrt(λ(s,mJpsi^2,mB^2))
    K00*(p*q)^j*(s+mJpsi^2-mB^2)/(2*mJpsi^2)*(
        sqrt((2*j-1)/(2*j+1))*ClebschGordon(j-1,0,1,0,j,0)*cS+
        λ(s,mJpsi^2,mB^2)*sqrt((2*j+3)/(2*j+1))*ClebschGordon(j+1,0,1,0,j,0)*cD)
end
function A11_our(s,cS,cD)
    j = 1
    lam=0
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    K10 = q
    K10*(p*q)^(j-1)*(
        sqrt((2*j-1)/(2*j+1))*ClebschGordon(j-1,0,1,1,j,1)*cS+
        λ(s,mJpsi^2,mB^2)*sqrt((2*j+3)/(2*j+1))*ClebschGordon(j+1,0,1,1,j,1)*cD)
end

Msq_our(s,zs,cS,cD) = 3*(abs(A10_our(s,cS,cD)*Wignerd(1,0,0,zs))^2+2*abs(A11_our(s,cS,cD)*Wignerd(1,1,0,zs))^2)

ph(s) = sqrt(λ(s,mπ^2,mK^2)*λ(s,mJpsi^2,mB^2))/s;
function project_general(s,cS,cD,A10,A11)
    ph(s)*(abs(A10(s,cS,cD))^2+2*abs(A11(s,cS,cD))^2)
end

project_general (generic function with 1 method)

In [37]:
s_th = (mπ+mK)^2+0.0001
s_en = (mB-mJpsi)^2-0.0001;

# first, our amplitude from tensor form
function project(s)
    ph(s)*quadgk(zs->Msq_our(s,zs,0,1),-0.999,0.999)[1]
end
norm1 = quadgk(s->project(s),s_th,s_en)[1]
plot(s->project(s)/norm1,s_th:0.03:s_en, lab="sum of helicities", title="JPAC model")

# second, numerical projection, helicity
function project(s)
    ph(s)*quadgk(zs->amplsp_analytic(s,zs,[[0,0,1]]),-0.999,0.999)[1]
end
norm1 = quadgk(s->project(s),s_th,s_en)[1]
plot!(s->project(s)/norm1,s_th:0.03:s_en, lab="full model with tensors")

# analytic integrals
norm1 = quadgk(s->project_general(s,0,1,A10_our,A11_our),s_th,s_en)[1]
plot!(s->project_general(s,0,1,A10_our,A11_our)/norm1,s_th:0.03:s_en, lab="analyt. cosT integr")

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.2 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 JPAC model 
 
<polyline clip-path="url(#clip9202)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,353.436 47.6613,318.727 55.381,281.032 63.1006,243.778 70.8202,208.663 78.5398,176.552 86.2594,147.845 93.979,122.671 101.699,100.998 
 109.418,82.6955 117.138,67.5813 124.858,55.442 132.577,46.0521 140.297,39.1841 148.016,34.6147 155.736,32.1295 163.456,31.5244 171.175,32.6076 178.895,35.1997 
 186.614,39.1338 194.334,44.255 202.054,50.4201 209.773,57.497 217.493,65.3638 225.213,73.9085 232.932,83.028 240.652,92.6277 248.371,102.621 256.091,112.927 
 263.811,123.474 271.53,134.194 279.25,145.026 286.969,155.915 294.689,166.809 302.409,177.661 310.128,188.429 317.848,199.074 325.568,209.562 333.287,219.859 
 341.007,229.939 348.726,239.774 356.446,249.342 364.166,258.622 371.885,267.596 379.605,276.248 387.325,284.564 395.044,292.532 402.764,300.141 410.483,307.384 
 418.203,314.253 425.923,320.743 433.642,326.851 441.362,332.575 449.081,337.914 456.801,342.867 464.521,347.439 472.24,351.63 479.96,355.448 487.68,358.896 
 495.399,361.983 503.119,364.718 510.838,367.11 518.558,369.171 526.278,370.915 533.997,372.356 541.717,373.511 549.436,374.401 557.156,375.046 564.876,375.474 
 572.595,375.715 580.315,375.809 
 "/>
<polyline clip-path="url(#clip9202)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,353.436 47.6613,318.727 55.381,281.032 63.1006,243.778 70.8202,208.663 78.5398,176.552 86.2594,147.845 93.979,122.671 101.699,100.998 
 109.418,82.6955 117.138,67.5813 124.858,55.442 132.577,46.0521 140.297,39.1841 148.016,34.6147 155.736,32.1295 163.456,31.5244 171.175,32.6076 178.895,35.1997 
 186.614,39.1338 194.334,44.255 202.054,50.4201 209.773,57.497 217.493,65.3638 225.213,73.9085 232.932,83.028 240.652,92.6277 248.371,102.621 256.091,112.927 
 263.811,123.474 271.53,134.194 279.25,145.026 286.969,155.915 294.689,166.809 302.409,177.661 310.128,188.429 317.848,199.074 325.568,209.562 333.287,219.859 
 341.007,229.939 348.726,239.774 356.446,249.342 364.166,258.622 371.885,267.596 379.605,276.248 387.325,284.564 395.044,292.532 402.764,300.141 410.483,307.384 
 418.203,314.253 425.923,320.743 433.642,326.851 441.362,332.575 449.081,337.914 456.801,342.867 464.521,347.439 472.24,351.63 479.96,355.448 487.68,358.896 
 495.399,361.983 503.119,364.718 510.838,367.11 518.558,369.171 526.278,370.915 533.997,372.356 541.717,373.511 549.436,374.401 557.156,375.046 564.876,375.474 
 572.595,375.715 580.315,375.809 
 "/>
<polyline clip-path="url(#clip9202)" style="stroke:#3da44d; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,353.43 47.6613,318.712 55.381,281.009 63.1006,243.746 70.8202,208.625 78.5398,176.509 86.2594,147.799 93.979,122.623 101.699,100.949 
 109.418,82.6468 117.138,67.5338 124.858,55.3964 132.577,46.0092 140.297,39.1443 148.016,34.5785 155.736,32.0971 163.456,31.4961 171.175,32.5835 178.895,35.1799 
 186.614,39.1182 194.334,44.2437 202.054,50.4129 209.773,57.4937 217.493,65.3644 225.213,73.9127 232.932,83.0356 240.652,92.6383 248.371,102.634 256.091,112.943 
 263.811,123.492 271.53,134.215 279.25,145.049 286.969,155.939 294.689,166.834 302.409,177.687 310.128,188.456 317.848,199.102 325.568,209.59 333.287,219.887 
 341.007,229.967 348.726,239.802 356.446,249.369 364.166,258.649 371.885,267.622 379.605,276.273 387.325,284.588 395.044,292.555 402.764,300.163 410.483,307.404 
 418.203,314.272 425.923,320.761 433.642,326.868 441.362,332.59 449.081,337.927 456.801,342.879 464.521,347.449 472.24,351.64 479.96,355.456 487.68,358.903 
 495.399,361.989 503.119,364.723 510.838,367.114 518.558,369.174 526.278,370.917 533.997,372.357 541.717,373.512 549.436,374.401 557.156,375.

In [38]:
# LS amplitude, scattering
function A1λ_LS(s,cS,cD,lam)
    j = 1
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    p^(j-1)*q^j*(
        sqrt((2*j-1)/(2*j+1))*ClebschGordon(j-1,0,1,lam,j,lam)*cS+
        sqrt((2*j+3)/(2*j+1))*ClebschGordon(j+1,0,1,lam,j,lam)*p^2*cD)
end
A10_LS(s,cS,cD) = A1λ_LS(s,cS,cD,0)
A11_LS(s,cS,cD) = A1λ_LS(s,cS,cD,1)

# LS amplitude, decay
function A1λ_LS_decay(s,cS,cD,lam)
    j = 1
    p = sqrt(λ(s,mJpsi^2,mB^2))/(2*mB)
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    p^(j-1)*q^j*(
        sqrt((2*j-1)/(2*j+1))*ClebschGordon(j-1,0,1,lam,j,lam)*cS+
        sqrt((2*j+3)/(2*j+1))*ClebschGordon(j+1,0,1,lam,j,lam)*p^2*cD)
end
A10_LS_decay(s,cS,cD) = A1λ_LS_decay(s,cS,cD,0)
A11_LS_decay(s,cS,cD) = A1λ_LS_decay(s,cS,cD,1)


# tensor amplitude, decay
function A10_ten(s,cS,cD)
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    γ = (mB^2+s-mJpsi^2)/(2*mB*sqrt(s))
    E1 = (s+mJpsi^2-mB^2)/(2*sqrt(s))
    q*(E1/mJpsi*cS - γ*p^2*s/mB^2*(s-mJpsi^2-mB^2)/(2*mJpsi*mB)*cD)
end
function A11_ten(s,cS,cD)
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    q*(cS + p^2/2*s/mB^2*cD)
end

# tensor amplitude, scattering
function A10_ten_sc(s,cS,cD)
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    E1 = (s+mJpsi^2-mB^2)/(2*sqrt(s))
    E1/mJpsi*q*(cS - p^2*cD)
end
function A11_ten_sc(s,cS,cD)
    p = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
    q = sqrt(λ(s,mπ^2,mK^2)/(4*s))
    q*(cS + p^2/2*cD)
end

A11_ten_sc (generic function with 1 method)

In [39]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our),
    s->project_general(s,0,1,A10_LS,A11_LS),
    s->project_general(s,0,1,A10_LS_decay,A11_LS_decay),
    s->project_general(s,0,1,A10_ten_sc,A11_ten_sc),
    s->project_general(s,0,1,A10_ten,A11_ten)
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.03:s_en,
    lab=["JPAC","LS-scatt.","LS-decay","CPT-scatt.","CPT-decay"],
    ls=[:solid :solid :dash :solid :dash],
    c=[:red :blue :blue :green :green],
    grid=false)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.2 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 1.2 
 
 
 1.4 
 
<polyline clip-path="url(#clip9402)" style="stroke:#ff0000; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.811 39.9417,360.689 47.6613,337.234 55.381,311.762 63.1006,286.588 70.8202,262.86 78.5398,241.163 86.2594,221.767 93.979,204.758 101.699,190.115 
 109.418,177.75 117.138,167.54 124.858,159.34 132.577,152.998 140.297,148.361 148.016,145.276 155.736,143.6 163.456,143.194 171.175,143.928 178.895,145.682 
 186.614,148.343 194.334,151.806 202.054,155.974 209.773,160.757 217.493,166.075 225.213,171.85 232.932,178.013 240.652,184.501 248.371,191.254 256.091,198.218 
 263.811,205.345 271.53,212.589 279.25,219.909 286.969,227.266 294.689,234.627 302.409,241.959 310.128,249.234 317.848,256.426 325.568,263.512 333.287,270.469 
 341.007,277.278 348.726,283.923 356.446,290.387 364.166,296.656 371.885,302.718 379.605,308.563 387.325,314.18 395.044,319.562 402.764,324.702 410.483,329.594 
 418.203,334.234 425.923,338.618 433.642,342.744 441.362,346.61 449.081,350.215 456.801,353.561 464.521,356.649 472.24,359.48 479.96,362.058 487.68,364.387 
 495.399,366.472 503.119,368.318 510.838,369.934 518.558,371.326 526.278,372.503 533.997,373.476 541.717,374.256 549.436,374.857 557.156,375.293 564.876,375.582 
 572.595,375.745 580.315,375.808 
 "/>
<polyline clip-path="url(#clip9402)" style="stroke:#0000ff; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.806 39.9417,340.539 47.6613,291.834 55.381,245.103 63.1006,204.442 70.8202,170.925 78.5398,144.425 86.2594,124.333 93.979,109.868 101.699,100.232 
 109.418,94.6706 117.138,92.5056 124.858,93.1416 132.577,96.0641 140.297,100.833 148.016,107.074 155.736,114.472 163.456,122.761 171.175,131.718 178.895,141.156 
 186.614,150.92 194.334,160.88 202.054,170.93 209.773,180.979 217.493,190.956 225.213,200.802 232.932,210.466 240.652,219.912 248.371,229.107 256.091,238.028 
 263.811,246.657 271.53,254.978 279.25,262.984 286.969,270.667 294.689,278.025 302.409,285.055 310.128,291.759 317.848,298.139 325.568,304.199 333.287,309.945 
 341.007,315.381 348.726,320.515 356.446,325.354 364.166,329.905 371.885,334.177 379.605,338.178 387.325,341.917 395.044,345.403 402.764,348.644 410.483,351.649 
 418.203,354.427 425.923,356.988 433.642,359.34 441.362,361.491 449.081,363.452 456.801,365.23 464.521,366.834 472.24,368.273 479.96,369.555 487.68,370.689 
 495.399,371.684 503.119,372.546 510.838,373.286 518.558,373.911 526.278,374.429 533.997,374.849 541.717,375.18 549.436,375.429 557.156,375.607 564.876,375.723 
 572.595,375.787 580.315,375.812 
 "/>
<polyline clip-path="url(#clip9402)" style="stroke:#0000ff; stroke-width:1; stroke-opacity:1; fill:none" stroke-dasharray="8, 5" points="
 32.2221,375.813 39.9417,370.07 47.6613,360.114 55.381,348.006 63.1006,334.651 70.8202,320.633 78.5398,306.372 86.2594,292.182 93.979,278.298 101.699,264.903 
 109.418,252.136 117.138,240.102 124.858,228.879 132.577,218.524 140.297,209.075 148.016,200.557 155.736,192.983 163.456,186.354 171.175,180.666 178.895,175.906 
 186.614,172.056 194.334,169.093 202.054,166.99 209.773,165.718 217.493,165.243 225.213,165.53 232.932,166.544 240.652,168.245 248.371,170.595 256.091,173.554 
 263.811,177.08 271.53,181.133 279.25,185.671 286.969,190.653 294.689,196.037 302.409,201.783 310.128,207.85 317.848,214.197 325.568,220.784 333.287,227.573 
 341.007,234.524 348.726,241.601 356.446,248.765 364.166,255.982 371.885,263.215 379.605,270.432 387.325,277.6 395.044,284.685 402.764,291.659 410.483,298.493 
 418.203,305.157 425.923,311.626 433.642,317.874 441.362,323.879 449.081,329.619 456.801,335.073 464.521,340.223 472.24,345.054 479.96,349.549 487.68,353.698 
 495.399,357.49 503.119,360.917 510.838,363.975 518.558,366.661 526.278,368.976 533.997,370.925 541.717,372.516 549.436,373.763 557.156,3

In [95]:
savefig("/tmp/energy-factors.pdf")

In [41]:
# Blatt-Weisskopf functions
BWs = [z->1./(1.+z), z->1./(9+3*z+z^2)]
Rbw=5; # /GeV
# s-channel break up momentu
p(s) = sqrt(λ(s,mJpsi^2,mB^2)/(4*s))
q(s) = sqrt(λ(s,mπ^2,mK^2)/(4*s))
ampsqBW(s) = abs(bw(s,mKs^2,ΓKs))^2*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2)

ampsqBW (generic function with 1 method)

In [43]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our)*ampsqBW(s),
    s->project_general(s,0,1,A10_LS,A11_LS)*ampsqBW(s),
    s->project_general(s,0,1,A10_LS_decay,A11_LS_decay)*ampsqBW(s),
    s->project_general(s,0,1,A10_ten_sc,A11_ten_sc)*ampsqBW(s),
    s->project_general(s,0,1,A10_ten,A11_ten)*ampsqBW(s)
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.01:s_en,
    lab=["JPAC","LS-scatt.","LS-decay","CPT-scatt.","CPT-decay"],
    ls=[:solid :solid :dash :solid :dash],
    c=[:red :blue :blue :green :green],
    xlab="\$M_{\\pi K}\\textrm{ (GeV)}\$", 
    ylab="Normalized differentian width", # d\\Gamma / d M_{\\pi K}^2
    title="with Blatt-Weisskopf factors",
    legend=:topleft,
    grid=false
)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.2 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 1.2 
 
 
 1.4 
 
 
 with Blatt-Weisskopf factors 
 
 
<image width="85" height="17" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAFUAAAARCAYAAACsPOAgAAABu0lEQVRYhe1X0ZGEIAwNV4E1aAdb
gyVYgyVYgyVYg5ZgC9oBtmAL7z5u2UFNEFDGu5ljJjO7CUmeDxJAAaD/ce/4ehpA7FBKibthGAYo
peCakxQPgIMQEbquA2cz9h9X3p5apNzTNLG4YvCe+Uh5ABxJraoKPiCeJLWqqkNuidAreF0+eZ7L
uUS2T3bDU6RKeVNsAiICt4BnPgdl27YiqVprJ+FPkaq1TrLI0uY5qwj1/sE2XgAq1JZ6cLnNIRGK
yY4lfROnX5YFRVGI3GxO/2EYxNPyjpN0HEcY4fRX44eMPVkAlPSNtn6eZ5FQIqJpmkjc1uToXZIt
tnxNy/GRsizZcgzFxM3ndK/XCz68bOJIgG3ndV03evt/6IfYvjFxOFJ9eureHrIIwQvmCpRlGbIs
iwbjAhcbx1VBfd+LMa+Savg4m6u1lo1N03iffHVdewFMSWqojZsr3T1NJXiVvn2l4hz2peYT2NUf
U5Mq2SXcKVrAhtSrYpK6ynP/9I0h1sen7/vPzort/3+G1Fige2ma5haiUsmthBoxL6/N6u1KsK7r
jy7kSnVlMZKT+cYlvqh++3jyVccNG883xRFjxOx4wygAAAAASUVORK5CYII=
" transform="translate(272, 380)"/>
 
 
 Normalized differentian width 
 
<polyline clip-path="url(#clip9802)" style="stroke:#ff0000; stroke-width:1; stroke-opacity:1; fill:none" points="
 47.9701,360.065 50.4577,360.029 52.9453,359.964 55.4329,359.88 57.9205,359.782 60.4081,359.671 62.8957,359.55 65.3833,359.419 67.8709,359.279 70.3585,359.131 
 72.8461,358.976 75.3337,358.813 77.8212,358.644 80.3088,358.468 82.7964,358.286 85.284,358.098 87.7716,357.903 90.2592,357.703 92.7468,357.497 95.2344,357.285 
 97.722,357.067 100.21,356.844 102.697,356.614 105.185,356.379 107.672,356.138 110.16,355.89 112.648,355.637 115.135,355.377 117.623,355.112 120.11,354.839 
 122.598,354.56 125.086,354.275 127.573,353.983 130.061,353.683 132.548,353.377 135.036,353.063 137.523,352.742 140.011,352.413 142.499,352.076 144.986,351.732 
 147.474,351.379 149.961,351.017 152.449,350.647 154.937,350.267 157.424,349.879 159.912,349.481 162.399,349.073 164.887,348.655 167.375,348.227 169.862,347.788 
 172.35,347.338 174.837,346.876 177.325,346.403 179.813,345.918 182.3,345.42 184.788,344.91 187.275,344.386 189.763,343.849 192.251,343.297 194.738,342.731 
 197.226,342.15 199.713,341.553 202.201,340.941 204.688,340.311 207.176,339.665 209.664,339 212.151,338.318 214.639,337.616 217.126,336.895 219.614,336.153 
 222.102,335.39 224.589,334.606 227.077,333.798 229.564,332.968 232.052,332.113 234.54,331.233 237.027,330.327 239.515,329.395 242.002,328.434 244.49,327.444 
 246.978,326.424 249.465,325.373 251.953,324.29 254.44,323.173 256.928,322.021 259.416,320.833 261.903,319.608 264.391,318.343 266.878,317.038 269.366,315.691 
 271.853,314.3 274.341,312.863 276.829,311.38 279.316,309.847 281.804,308.263 284.291,306.626 286.779,304.934 289.267,303.185 291.754,301.376 294.242,299.506 
 296.729,297.57 299.217,295.568 301.705,293.497 304.192,291.353 306.68,289.134 309.167,286.837 311.655,284.459 314.143,281.996 316.63,279.447 319.118,276.806 
 321.605,274.072 324.093,271.24 326.581,268.307 329.068,265.269 331.556,262.123 334.043,258.865 336.531,255.492 339.018,252 341.506,248.385 343.994,244.644 
 346.481,240.773 348.969,236.769 351.456,232.63 353.944,228.351 356.432,223.932 358.919,219.37 361.407,214.663 363.894,209.811 366.382,204.813 368.87,199.67 
 371.357,194.384 373.845,188.956 376.332,183.391 378.82,177.694 381.308,171.872 383.795,165.933 386.283,159.888 388.77,153.748 391.258,147.53 393.746,141.249 
 396.233,134.927 398.721,128.584 401.208,122.247 403.696,115.944 406.183,109.706 408.671,103.566 411.159,97.5609 413.646,91.7299 416.134,86.1137 418.621,80.755 
 421.109,75.6974 423.597,70.9855 426.084,66.6631 428.572,62.7736 431.059,59.3582 433.547,56.4555 436.035,54.1006 438.522,52.324 441.01,51.1512 443.497,50.6017 
 445.985,50.6884 448.473,51.4175 450.96,52.7881 453.448,54.7924 455.935,57.4154 458.423,60.6359 460.911,64.4266 463.398,68.7549 465.886,73.5837 468.373,78.8725 
 470.861,84.5777 473.348,90.6544 475.836,97.0563 478.324,103.738 480.811,110.653 483.299,117.757 485.786,125.01

In [36]:
savefig("/tmp/noBW.pdf")

In [37]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our)*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2),
    s->project_general(s,0,1,A10_LS,A11_LS)*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2),
    s->project_general(s,0,1,A10_ten,A11_ten)*BWs[1]((Rbw*q(s))^2)*BWs[2]((Rbw*p(s))^2)
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.03:s_en, lab=["JPAC","LS","tensor"])

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.1 
 
 
 0.2 
 
 
 0.3 
 
 
 0.4 
 
 
 0.5 
 
 
 0.6 
 
 
 0.7 
 
<polyline clip-path="url(#clip0102)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,370.931 47.6613,362.718 55.381,352.962 63.1006,342.346 70.8202,331.249 78.5398,319.9 86.2594,308.445 93.979,296.981 101.699,285.576 
 109.418,274.276 117.138,263.113 124.858,252.113 132.577,241.294 140.297,230.668 148.016,220.248 155.736,210.042 163.456,200.057 171.175,190.299 178.895,180.774 
 186.614,171.487 194.334,162.442 202.054,153.645 209.773,145.098 217.493,136.808 225.213,128.777 232.932,121.012 240.652,113.516 248.371,106.294 256.091,99.3514 
 263.811,92.6937 271.53,86.3264 279.25,80.2554 286.969,74.4871 294.689,69.0282 302.409,63.8858 310.128,59.0674 317.848,54.5812 325.568,50.4359 333.287,46.6407 
 341.007,43.2056 348.726,40.1412 356.446,37.4592 364.166,35.1721 371.885,33.2932 379.605,31.8374 387.325,30.8206 395.044,30.2603 402.764,30.1757 410.483,30.5879 
 418.203,31.5203 425.923,32.9989 433.642,35.0527 441.362,37.7143 449.081,41.0205 456.801,45.0135 464.521,49.7415 472.24,55.2603 479.96,61.6355 487.68,68.9447 
 495.399,77.2812 503.119,86.759 510.838,97.5199 518.558,109.745 526.278,123.67 533.997,139.617 541.717,158.041 549.436,179.625 557.156,205.482 564.876,237.624 
 572.595,280.316 580.315,343.12 
 "/>
<polyline clip-path="url(#clip0102)" style="stroke:#e26f46; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.811 39.9417,358.965 47.6613,333.642 55.381,306.825 63.1006,280.723 70.8202,256.229 78.5398,233.672 86.2594,213.116 93.979,194.501 101.699,177.708 
 109.418,162.597 117.138,149.026 124.858,136.858 132.577,125.964 140.297,116.228 148.016,107.545 155.736,99.8209 163.456,92.9721 171.175,86.9238 178.895,81.6094 
 186.614,76.9698 194.334,72.9523 202.054,69.51 209.773,66.601 217.493,64.188 225.213,62.2376 232.932,60.7199 240.652,59.6083 248.371,58.8787 256.091,58.51 
 263.811,58.4829 271.53,58.7803 279.25,59.387 286.969,60.2894 294.689,61.4754 302.409,62.9345 310.128,64.6572 317.848,66.6355 325.568,68.8623 333.287,71.3319 
 341.007,74.0392 348.726,76.9806 356.446,80.153 364.166,83.5547 371.885,87.1846 379.605,91.0429 387.325,95.1307 395.044,99.4502 402.764,104.005 410.483,108.799 
 418.203,113.839 425.923,119.132 433.642,124.687 441.362,130.516 449.081,136.632 456.801,143.051 464.521,149.794 472.24,156.883 479.96,164.348 487.68,172.223 
 495.399,180.552 503.119,189.389 510.838,198.802 518.558,208.881 526.278,219.746 533.997,231.56 541.717,244.559 549.436,259.104 557.156,275.787 564.876,295.693 
 572.595,321.141 580.315,357.329 
 "/>
<polyline clip-path="url(#clip0102)" style="stroke:#3da44d; stroke-width:1; stroke-opacity:1; fill:none" points="
 32.2221,375.813 39.9417,372.171 47.6613,365.957 55.381,358.464 63.1006,350.185 70.8202,341.396 78.5398,332.265 86.2594,322.903 93.979,313.385 101.699,303.766 
 109.418,294.083 117.138,284.366 124.858,274.638 132.577,264.917 140.297,255.219 148.016,245.557 155.736,235.942 163.456,226.385 171.175,216.897 178.895,207.486 
 186.614,198.162 194.334,188.933 202.054,179.809 209.773,170.799 217.493,161.912 225.213,153.156 232.932,144.542 240.652,136.078 248.371,127.776 256.091,119.644 
 263.811,111.694 271.53,103.937 279.25,96.3837 286.969,89.0464 294.689,81.9373 302.409,75.0694 310.128,68.4563 317.848,62.112 325.568,56.0516 333.287,50.2908 
 341.007,44.8461 348.726,39.7352 356.446,34.9768 364.166,30.5905 371.885,26.5977 379.605,23.021 387.325,19.8848 395.044,17.2152 402.764,15.0408 410.483,13.3922 
 418.203,12.3033 425.923,11.811 433.642,11.9561 441.362,12.7838 449.081,14.3448 456.801,16.6961 464.521,19.9023 472.24,24.0377 479.96,29.1883 487.68,35.455 
 495.399,42.958 503.119,51.8425 510.838,62.2878 518.558,74.5205 526.278,88.8347 533.997,105.626 541.717,125.453 549.436,149.146 557.

In [50]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_LS,A11_LS)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_LS_decay,A11_LS_decay)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_ten_sc,A11_ten_sc)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_ten,A11_ten)*abs(bw(s,mKs^2,ΓKs))^2
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.01:s_en,
    lab=["JPAC","LS-scatt.","LS-decay","CPT-scatt.","CPT-decay"],
    ls=[:solid :solid :dash :solid :dash],
    c=[:red :blue :blue :green :green],
    xlab="\$M_{\\pi K}\\textrm{ (GeV)}\$",
    ylab="Normalized differentian width", # d\\Gamma / d M_{\\pi K}^2
    title="without Blatt-Weisskopf factors",
    legend=:topleft,
    grid=false
)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.2 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 without Blatt-Weisskopf factors 
 
 
<image width="85" height="17" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAFUAAAARCAYAAACsPOAgAAABu0lEQVRYhe1X0ZGEIAwNV4E1aAdb
gyVYgyVYgyVYg5ZgC9oBtmAL7z5u2UFNEFDGu5ljJjO7CUmeDxJAAaD/ce/4ehpA7FBKibthGAYo
peCakxQPgIMQEbquA2cz9h9X3p5apNzTNLG4YvCe+Uh5ABxJraoKPiCeJLWqqkNuidAreF0+eZ7L
uUS2T3bDU6RKeVNsAiICt4BnPgdl27YiqVprJ+FPkaq1TrLI0uY5qwj1/sE2XgAq1JZ6cLnNIRGK
yY4lfROnX5YFRVGI3GxO/2EYxNPyjpN0HEcY4fRX44eMPVkAlPSNtn6eZ5FQIqJpmkjc1uToXZIt
tnxNy/GRsizZcgzFxM3ndK/XCz68bOJIgG3ndV03evt/6IfYvjFxOFJ9eureHrIIwQvmCpRlGbIs
iwbjAhcbx1VBfd+LMa+Savg4m6u1lo1N03iffHVdewFMSWqojZsr3T1NJXiVvn2l4hz2peYT2NUf
U5Mq2SXcKVrAhtSrYpK6ynP/9I0h1sen7/vPzort/3+G1Fige2ma5haiUsmthBoxL6/N6u1KsK7r
jy7kSnVlMZKT+cYlvqh++3jyVccNG883xRFjxOx4wygAAAAASUVORK5CYII=
" transform="translate(272, 380)"/>
 
 
 Normalized differentian width 
 
<polyline clip-path="url(#clip0602)" style="stroke:#ff0000; stroke-width:1; stroke-opacity:1; fill:none" points="
 47.9701,360.065 50.4577,359.236 52.9453,357.787 55.4329,355.979 57.9205,353.915 60.4081,351.658 62.8957,349.251 65.3833,346.728 67.8709,344.113 70.3585,341.427 
 72.8461,338.687 75.3337,335.906 77.8212,333.095 80.3088,330.264 82.7964,327.421 85.284,324.573 87.7716,321.726 90.2592,318.884 92.7468,316.051 95.2344,313.232 
 97.722,310.428 100.21,307.643 102.697,304.879 105.185,302.136 107.672,299.417 110.16,296.723 112.648,294.055 115.135,291.413 117.623,288.798 120.11,286.21 
 122.598,283.65 125.086,281.117 127.573,278.612 130.061,276.134 132.548,273.684 135.036,271.261 137.523,268.865 140.011,266.496 142.499,264.152 144.986,261.835 
 147.474,259.543 149.961,257.275 152.449,255.032 154.937,252.812 157.424,250.616 159.912,248.442 162.399,246.291 164.887,244.161 167.375,242.051 169.862,239.962 
 172.35,237.893 174.837,235.842 177.325,233.81 179.813,231.795 182.3,229.798 184.788,227.817 187.275,225.852 189.763,223.901 192.251,221.966 194.738,220.044 
 197.226,218.135 199.713,216.239 202.201,214.355 204.688,212.483 207.176,210.62 209.664,208.768 212.151,206.926 214.639,205.092 217.126,203.266 219.614,201.448 
 222.102,199.637 224.589,197.832 227.077,196.033 229.564,194.239 232.052,192.45 234.54,190.664 237.027,188.882 239.515,187.103 242.002,185.325 244.49,183.55 
 246.978,181.775 249.465,180.001 251.953,178.227 254.44,176.453 256.928,174.677 259.416,172.9 261.903,171.12 264.391,169.338 266.878,167.553 269.366,165.765 
 271.853,163.972 274.341,162.176 276.829,160.375 279.316,158.568 281.804,156.757 284.291,154.94 286.779,153.117 289.267,151.289 291.754,149.454 294.242,147.613 
 296.729,145.766 299.217,143.913 301.705,142.054 304.192,140.19 306.68,138.321 309.167,136.446 311.655,134.568 314.143,132.685 316.63,130.801 319.118,128.914 
 321.605,127.028 324.093,125.142 326.581,123.259 329.068,121.38 331.556,119.508 334.043,117.645 336.531,115.794 339.018,113.958 341.506,112.14 343.994,110.344 
 346.481,108.575 348.969,106.837 351.456,105.135 353.944,103.476 356.432,101.866 358.919,100.313 361.407,98.8228 363.894,97.4055 366.382,96.0699 368.87,94.8263 
 371.357,93.6856 373.845,92.6595 376.332,91.7608 378.82,91.0031 381.308,90.4007 383.795,89.969 386.283,89.7241 388.77,89.6828 391.258,89.8626 393.746,90.2815 
 396.233,90.9579 398.721,91.9103 401.208,93.1572 403.696,94.7166 406.183,96.6062 408.671,98.8423 411.159,101.44 413.646,104.413 416.134,107.771 418.621,111.523 
 421.109,115.675 423.597,120.227 426.084,125.178 428.572,130.521 431.059,136.243 433.547,142.33 436.035,148.761 438.522,155.511 441.01,162.549 443.497,169.843 
 445.985,177.355 448.473,185.045 450.96,192.871 453.448,200.789 455.935,208.756 458.423,216.726 460.911,224.657 463.398,232.508 465.886,240.238 468.373,247.814 
 470.861,255.201 473.348,262.371 475.836,269.299 478.324,275.965 480.811,282.351 483.299,288.446 485.786,294.239 488.274

In [51]:
savefig("/tmp/withoutBW.pdf")

In [38]:
funcs=[
    s->project_general(s,0,1,A10_our,A11_our)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_LS,A11_LS)*abs(bw(s,mKs^2,ΓKs))^2,
    s->project_general(s,0,1,A10_ten,A11_ten)*abs(bw(s,mKs^2,ΓKs))^2
    ]
norms = [quadgk(s->f(s),s_th,s_en)[1] for f in funcs]
nfuncs = [s->funcs[i](s)/norms[i] for i=1:length(funcs)]
plot(nfuncs, s_th:0.01:s_en, lab=["JPAC","LS","CPT"],
    xlab="\$M_{\\pi K}\\textrm{ (GeV)}\$", ylab="Differentian cross section",
    title="no Blatt-Weisskopf factors",
    legend=:topleft,
    c=:black,
ls=[:solid :dash :dashdot])

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.2 
 
 
 0.4 
 
 
 0.6 
 
 
 0.8 
 
 
 1.0 
 
 
 no Blatt-Weisskopf factors 
 
 
<image width="85" height="17" xlink:href="data:;base64,
iVBORw0KGgoAAAANSUhEUgAAAFUAAAARCAYAAACsPOAgAAABu0lEQVRYhe1X0ZGEIAwNV4E1aAdb
gyVYgyVYgyVYg5ZgC9oBtmAL7z5u2UFNEFDGu5ljJjO7CUmeDxJAAaD/ce/4ehpA7FBKibthGAYo
peCakxQPgIMQEbquA2cz9h9X3p5apNzTNLG4YvCe+Uh5ABxJraoKPiCeJLWqqkNuidAreF0+eZ7L
uUS2T3bDU6RKeVNsAiICt4BnPgdl27YiqVprJ+FPkaq1TrLI0uY5qwj1/sE2XgAq1JZ6cLnNIRGK
yY4lfROnX5YFRVGI3GxO/2EYxNPyjpN0HEcY4fRX44eMPVkAlPSNtn6eZ5FQIqJpmkjc1uToXZIt
tnxNy/GRsizZcgzFxM3ndK/XCz68bOJIgG3ndV03evt/6IfYvjFxOFJ9eureHrIIwQvmCpRlGbIs
iwbjAhcbx1VBfd+LMa+Savg4m6u1lo1N03iffHVdewFMSWqojZsr3T1NJXiVvn2l4hz2peYT2NUf
U5Mq2SXcKVrAhtSrYpK6ynP/9I0h1sen7/vPzort/3+G1Fige2ma5haiUsmthBoxL6/N6u1KsK7r
jy7kSnVlMZKT+cYlvqh++3jyVccNG883xRFjxOx4wygAAAAASUVORK5CYII=
" transform="translate(272, 380)"/>
 
 
 Differentian cross section 
 
<polyline clip-path="url(#clip0302)" style="stroke:#000000; stroke-width:1; stroke-opacity:1; fill:none" points="
 47.9701,360.065 50.4577,359.12 52.9453,357.466 55.4329,355.402 57.9205,353.046 60.4081,350.47 62.8957,347.723 65.3833,344.843 67.8709,341.859 70.3585,338.794 
 72.8461,335.666 75.3337,332.492 77.8212,329.284 80.3088,326.053 82.7964,322.808 85.284,319.558 87.7716,316.308 90.2592,313.064 92.7468,309.831 95.2344,306.613 
 97.722,303.413 100.21,300.235 102.697,297.079 105.185,293.95 107.672,290.847 110.16,287.772 112.648,284.726 115.135,281.711 117.623,278.726 120.11,275.773 
 122.598,272.85 125.086,269.96 127.573,267.101 130.061,264.273 132.548,261.477 135.036,258.711 137.523,255.976 140.011,253.272 142.499,250.598 144.986,247.952 
 147.474,245.336 149.961,242.748 152.449,240.188 154.937,237.655 157.424,235.148 159.912,232.667 162.399,230.212 164.887,227.781 167.375,225.373 169.862,222.989 
 172.35,220.627 174.837,218.286 177.325,215.967 179.813,213.668 182.3,211.388 184.788,209.127 187.275,206.884 189.763,204.658 192.251,202.449 194.738,200.255 
 197.226,198.077 199.713,195.913 202.201,193.763 204.688,191.625 207.176,189.5 209.664,187.386 212.151,185.283 214.639,183.19 217.126,181.107 219.614,179.031 
 222.102,176.964 224.589,174.904 227.077,172.851 229.564,170.804 232.052,168.761 234.54,166.723 237.027,164.689 239.515,162.659 242.002,160.63 244.49,158.604 
 246.978,156.578 249.465,154.553 251.953,152.529 254.44,150.503 256.928,148.477 259.416,146.448 261.903,144.417 264.391,142.383 266.878,140.346 269.366,138.305 
 271.853,136.259 274.341,134.209 276.829,132.153 279.316,130.092 281.804,128.024 284.291,125.95 286.779,123.87 289.267,121.783 291.754,119.689 294.242,117.588 
 296.729,115.48 299.217,113.365 301.705,111.244 304.192,109.116 306.68,106.982 309.167,104.843 311.655,102.699 314.143,100.551 316.63,98.3995 319.118,96.2466 
 321.605,94.0932 324.093,91.9409 326.581,89.7917 329.068,87.6476 331.556,85.511 334.043,83.3848 336.531,81.272 339.018,79.1761 341.506,77.1012 343.994,75.0516 
 346.481,73.0322 348.969,71.0486 351.456,69.1067 353.944,67.2134 356.432,65.3759 358.919,63.6026 361.407,61.9022 363.894,60.2845 366.382,58.7602 368.87,57.3409 
 371.357,56.0389 373.845,54.8678 376.332,53.8421 378.82,52.9773 381.308,52.2898 383.795,51.7971 386.283,51.5175 388.77,51.4704 391.258,51.6756 393.746,52.1537 
 396.233,52.9257 398.721,54.0127 401.208,55.4358 403.696,57.2157 406.183,59.3723 408.671,61.9244 411.159,64.8892 413.646,68.2819 416.134,72.1149 418.621,76.3977 
 421.109,81.1361 423.597,86.3319 426.084,91.9823 428.572,98.0796 431.059,104.611 433.547,111.558 436.035,118.898 438.522,126.601 441.01,134.635 443.497,142.959 
 445.985,151.533 448.473,160.31 450.96,169.242 453.448,178.28 455.935,187.372 458.423,196.469 460.911,205.52 463.398,214.48 465.886,223.304 468.373,231.95 
 470.861,240.381 473.348,248.564 475.836,256.472 478.324,264.08 480.811,271.369 483.299,278.324 485.786,284.937

In [39]:
savefig("/tmp/withBW.pdf")

In [40]:
adam_data = readdlm("fort.13");

In [41]:
normB = sum(adam_data[:,2])
dx = adam_data[2,1]-adam_data[1,1]
bar(adam_data[:,1],adam_data[:,2]/normB/dx, lab="Adam")
plot!(nfuncs[3], s_th:0.01:s_en, lab="Misha", c=:black,
    xlab="MpiK(GeV)", ylab="Differentian cross section",
    title="no Blatt-Weisskopf factors",
    legend=:topleft)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
 
 2.0 
 
 
 2.5 
 
 
 0.00 
 
 
 0.25 
 
 
 0.50 
 
 
 0.75 
 
 
 1.00 
 
 
 no Blatt-Weisskopf factors 
 
 
 MpiK(GeV) 
 
 
 Differentian cross section 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
<polyline clip-path="url(#clip0502)" style="stroke:#000000; stroke-width:1; stroke-opacity:1; fill:none" points="
 65.8306,360.065 68.161,359.367 70.4913,358.142 72.8217,356.605 75.1521,354.843 77.4824,352.905 79.8128,350.829 82.1431,348.64 84.4735,346.361 86.8038,344.007 
 89.1342,341.593 91.4645,339.13 93.7949,336.628 96.1253,334.094 98.4556,331.536 100.786,328.959 103.116,326.368 105.447,323.769 107.777,321.163 110.107,318.556 
 112.438,315.948 114.768,313.344 117.098,310.743 119.429,308.15 121.759,305.564 124.09,302.986 126.42,300.419 128.75,297.863 131.081,295.318 133.411,292.785 
 135.741,290.265 138.072,287.757 140.402,285.263 142.732,282.782 145.063,280.313 147.393,277.859 149.723,275.417 152.054,272.988 154.384,270.572 156.714,268.169 
 159.045,265.779 161.375,263.401 163.706,261.034 166.036,258.68 168.366,256.336 170.697,254.004 173.027,251.682 175.357,249.37 177.688,247.067 180.018,244.775 
 182.348,242.49 184.679,240.215 187.009,237.947 189.339,235.686 191.67,233.433 194,231.186 196.331,228.945 198.661,226.71 200.991,224.479 203.322,222.253 
 205.652,220.031 207.982,217.812 210.313,215.596 212.643,213.382 214.973,211.17 217.304,208.96 219.634,206.75 221.964,204.54 224.295,202.33 226.625,200.119 
 228.956,197.906 231.286,195.691 233.616,193.474 235.947,191.253 238.277,189.029 240.607,186.801 242.938,184.567 245.268,182.328 247.598,180.083 249.929,177.832 
 252.259,175.573 254.589,173.306 256.92,171.032 259.25,168.748 261.58,166.456 263.911,164.153 266.241,161.84 268.572,159.516 270.902,157.181 273.232,154.834 
 275.563,152.474 277.893,150.102 280.223,147.717 282.554,145.318 284.884,142.905 287.214,140.478 289.545,138.037 291.875,135.581 294.205,133.11 296.536,130.625 
 298.866,128.124 301.197,125.608 303.527,123.078 305.857,120.533 308.188,117.975 310.518,115.402 312.848,112.816 315.179,110.219 317.509,107.609 319.839,104.99 
 322.17,102.361 324.5,99.7257 326.83,97.0844 329.161,94.4398 331.491,91.7942 333.822,89.1504 336.152,86.5115 338.482,83.8811 340.813,81.2631 343.143,78.6621 
 345.473,76.0831 347.804,73.5318 350.134,71.0144 352.464,68.5378 354.795,66.1098 357.125,63.7388 359.455,61.4341 361.786,59.2061 364.116,57.0659 366.446,55.0257 
 368.777,53.0989 371.107,51.2997 373.438,49.6436 375.768,48.1473 378.098,46.8285 380.429,45.7061 382.759,44.8 385.089,44.1312 387.42,43.7217 389.75,43.5939 
 392.08,43.7712 394.411,44.2772 396.741,45.1355 399.071,46.3696 401.402,48.0023 403.732,50.0554 406.063,52.5492 408.393,55.5019 410.723,58.9291 413.054,62.8434 
 415.384,67.2535 417.714,72.164 420.045,77.5746 422.375,83.48 424.705,89.8691 427.036,96.7254 429.366,104.026 431.696,111.743 434.027,119.842 436.357,128.285 
 438.688,137.027 441.018,146.021 443.348,155.217 445.679,164.561 448.009,174.001 450.339,183.481 452.67,192.948 455,202.352 457.33,211.643 459.661,220.775 
 461.991,229.708 464.321,238.404 466.652,246.83 468.982,254.959 471.312,262.768 473.643,270.239 475.973,277.36 478.304,284.12 480.634,290.515 482.964,296.545 
 485.295,302.21 487.625,307.517 489.955,312.471 492.286,317.083 494.616,321.364 496.946,325.325 499.277,328.981 501.607,332.345 503.937,335.431 506.268,338.255 
 508.598,340.831 510.929,343.174 513.259,345.299 515.589,347.219 517.92,348.949 520.25,350.501 522.58,351.889 524.911,353.124 527.241,354.219 529.571,355.184 
 531.902,356.03 534.232,356.767 536.562,357.404 538.893,357.95 541.223,358.414 543.553,358.803 545.884,359.125 548.214,359.388 550.545,359.596 552.875,359.758 
 555.205,359.878 557.536,359.963 

In [42]:
savefig("/tmp/adam_tensor.pdf")